// @lifeomic/chroma-react/components/Menu/MenuItem.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MenuItem = exports.useStyles = exports.MenuItemStylesKey = void 0;
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
var styles_1 = require("../../styles");
exports.MenuItemStylesKey = 'ChromaMenuItem';
exports.useStyles = (0, styles_1.makeStyles)(function (theme) { return ({
    root: {
        alignItems: 'center',
        backgroundColor: theme.palette.common.white,
        border: 'none',
        cursor: 'pointer',
        display: 'flex',
        fontFamily: theme.typography.fontFamily,
        fontSize: theme.pxToRem(14),
        letterSpacing: '0.015625em',
        margin: 0,
        outline: 'none',
        overflow: 'hidden',
        padding: theme.spacing(1, 2),
        userSelect: 'none',
        width: '100%',
        '&:hover,&:focus': {
            backgroundColor: 'rgba(222,244,252, 0.6)',
        },
        '&:focus': {
            outline: 'none',
        },
    },
    icon: {
        marginRight: theme.spacing(1),
    },
}); }, { name: exports.MenuItemStylesKey });
exports.MenuItem = React.forwardRef(function (_a, ref) {
    var className = _a.className, children = _a.children, Icon = _a.icon, onClick = _a.onClick, text = _a.text, rootProps = __rest(_a, ["className", "children", "icon", "onClick", "text"]);
    var classes = (0, exports.useStyles)({});
    var handleStopPropagation = function (e) {
        e.stopPropagation();
        onClick && onClick(e);
    };
    return (React.createElement("button", __assign({ ref: ref, className: (0, clsx_1.default)(classes.root, className), onClick: handleStopPropagation }, rootProps),
        !!Icon && (React.createElement(Icon, { role: "img", "aria-hidden": true, className: classes.icon, width: 18, height: 18 })),
        !!text && text,
        children));
});
