// @lifeomic/chroma-react/components/Menu/MenuButton.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MenuButton = exports.useStyles = exports.MenuButtonStylesKey = void 0;
var React = __importStar(require("react"));
var styles_1 = require("../../styles");
var Button_1 = require("../Button");
exports.MenuButtonStylesKey = 'ChromaMenuButton';
exports.useStyles = (0, styles_1.makeStyles)(function (theme) { return ({
    trailingIcon: {
        width: theme.spacing(2),
        height: theme.spacing(2),
        marginLeft: theme.spacing(1),
        verticalAlign: 'middle',
        position: 'relative',
        top: theme.pxToRem(-1),
    },
}); }, { name: exports.MenuButtonStylesKey });
/** @deprecated Please use `trailingIcon` from Button instead. This will be removed in a future release */
exports.MenuButton = React.forwardRef(function (_a, ref) {
    var children = _a.children, TrailingIcon = _a.trailingIcon, rootProps = __rest(_a, ["children", "trailingIcon"]);
    var classes = (0, exports.useStyles)({});
    return (React.createElement(Button_1.Button, __assign({ ref: ref }, rootProps),
        children,
        !!TrailingIcon && (React.createElement(TrailingIcon, { role: "img", "aria-hidden": true, className: classes.trailingIcon }))));
});
