// @lifeomic/chroma-react/components/Menu/MenuButton.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var styles_1 = require("../../styles");
var Button_1 = require("../Button");
exports.MenuButtonStylesKey = 'ChromaMenuButton';
exports.useStyles = styles_1.makeStyles(function (theme) { return ({
    trailingIcon: {
        width: theme.spacing(2),
        height: theme.spacing(2),
        marginLeft: theme.spacing(1),
    },
}); }, { name: exports.MenuButtonStylesKey });
exports.MenuButton = React.forwardRef(function (_a, ref) {
    var children = _a.children, TrailingIcon = _a.trailingIcon, rootProps = __rest(_a, ["children", "trailingIcon"]);
    var classes = exports.useStyles({});
    return (React.createElement(Button_1.Button, __assign({ ref: ref }, rootProps),
        children,
        !!TrailingIcon && (React.createElement(TrailingIcon, { role: "img", "aria-hidden": true, className: classes.trailingIcon }))));
});
