// @lifeomic/chroma-react/components/Menu/Menu.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Menu = exports.useStyles = exports.MenuStylesKey = void 0;
var ConditionalWrapper_1 = require("../_private/ConditionalWrapper");
var styles_1 = require("../../styles");
var MenuItem_1 = require("./MenuItem");
var framer_motion_1 = require("framer-motion");
var Portal_1 = require("reakit/Portal");
var Text_1 = require("../Text");
var Menu_1 = require("reakit/Menu");
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
require("focus-visible");
exports.MenuStylesKey = 'ChromaMenu';
exports.useStyles = (0, styles_1.makeStyles)(function (theme) { return ({
    root: {
        backgroundColor: theme.palette.common.white,
        borderRadius: theme.pxToRem(10),
        boxShadow: theme.boxShadows.popover,
        maxHeight: theme.pxToRem(432),
        minWidth: theme.pxToRem(224),
        overflowY: 'auto',
        paddingBottom: theme.spacing(1),
        paddingTop: theme.spacing(1),
        zIndex: 40,
        '&:focus': {
            outline: 'none',
        },
        '&:focus.focus-visible': {
            boxShadow: theme.boxShadows.focusVisible,
        },
    },
    title: {
        paddingBottom: theme.spacing(2),
        paddingLeft: theme.spacing(2.5),
        paddingRight: theme.spacing(2.5),
    },
}); }, { name: exports.MenuStylesKey });
var isMenuItemElement = function (el) { return el.type === MenuItem_1.MenuItem; };
var Menu = function (_a) {
    var ariaLabel = _a["aria-label"], anchorElement = _a.anchorElement, children = _a.children, className = _a.className, gutter = _a.gutter, items = _a.items, _b = _a.placement, placement = _b === void 0 ? 'bottom' : _b, title = _a.title, _c = _a.usePortal, usePortal = _c === void 0 ? false : _c, rootProps = __rest(_a, ['aria-label', "anchorElement", "children", "className", "gutter", "items", "placement", "title", "usePortal"]);
    var classes = (0, exports.useStyles)({});
    var shouldReduceMotion = (0, framer_motion_1.useReducedMotion)();
    var menu = (0, Menu_1.useMenuState)({ placement: placement, gutter: gutter });
    var handleStopPropagation = function (e) {
        return e.stopPropagation();
    };
    return (React.createElement(React.Fragment, null,
        React.createElement(Menu_1.MenuButton, __assign({}, menu, anchorElement.props, { onClick: handleStopPropagation }), function (disclosureProps) {
            return React.cloneElement(React.Children.only(anchorElement), disclosureProps);
        }),
        React.createElement(ConditionalWrapper_1.ConditionalWrapper, { condition: usePortal, wrapper: function (children) { return React.createElement(Portal_1.Portal, null, children); } },
            React.createElement(Menu_1.Menu, __assign({}, menu, { as: framer_motion_1.motion.div, "aria-label": ariaLabel, className: (0, clsx_1.default)(classes.root, className), animate: menu.visible
                    ? shouldReduceMotion
                        ? { opacity: 1 }
                        : {
                            opacity: 1,
                            transition: { duration: 0.3 },
                        }
                    : shouldReduceMotion
                        ? { opacity: 0 }
                        : {
                            opacity: 0,
                            transition: { duration: 0.1 },
                        } }, rootProps),
                !!title && (React.createElement(Text_1.Text, { className: classes.title, weight: "bold", size: "subbody" }, title)),
                items &&
                    items.map(function (item, i) {
                        return isMenuItemElement(item) ? (React.createElement(Menu_1.MenuItem, __assign({}, menu, item.props, { key: "item-".concat(i), onClick: function (e) {
                                menu.hide();
                                item.props.onClick && item.props.onClick(e);
                            } }), function (itemProps) {
                            return React.cloneElement(React.Children.only(item), itemProps);
                        })) : (item);
                    })))));
};
exports.Menu = Menu;
