// @itwin/itwinui-react/cjs/core/ErrorPage/ErrorPage.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ErrorPage = void 0;
/*---------------------------------------------------------------------------------------------
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
var _401_1 = __importDefault(require("@itwin/itwinui-illustrations-react/cjs/illustrations/401"));
var _403_1 = __importDefault(require("@itwin/itwinui-illustrations-react/cjs/illustrations/403"));
var _404_1 = __importDefault(require("@itwin/itwinui-illustrations-react/cjs/illustrations/404"));
var _500_1 = __importDefault(require("@itwin/itwinui-illustrations-react/cjs/illustrations/500"));
var _502_1 = __importDefault(require("@itwin/itwinui-illustrations-react/cjs/illustrations/502"));
var _503_1 = __importDefault(require("@itwin/itwinui-illustrations-react/cjs/illustrations/503"));
var Error_1 = __importDefault(require("@itwin/itwinui-illustrations-react/cjs/illustrations/Error"));
var react_1 = __importDefault(require("react"));
var Button_1 = require("../Buttons/Button");
var utils_1 = require("../utils");
var classnames_1 = __importDefault(require("classnames"));
require("@itwin/itwinui-css/css/non-ideal-state.css");
/**
 * A stylized display to communicate common http errors.
 * @example
 * <ErrorPage errorType='401' />
 */
var ErrorPage = function (props) {
    var errorType = props.errorType, errorName = props.errorName, errorMessage = props.errorMessage, primaryButtonHandle = props.primaryButtonHandle, primaryButtonLabel = props.primaryButtonLabel, secondaryButtonHandle = props.secondaryButtonHandle, secondaryButtonLabel = props.secondaryButtonLabel, translatedErrorMessages = props.translatedErrorMessages, className = props.className, rest = __rest(props, ["errorType", "errorName", "errorMessage", "primaryButtonHandle", "primaryButtonLabel", "secondaryButtonHandle", "secondaryButtonLabel", "translatedErrorMessages", "className"]);
    (0, utils_1.useTheme)();
    var defaultErrorMessages = __assign({ badGateway: 'Bad gateway', error: 'Error', forbidden: 'Forbidden', internalServerError: 'Internal server error', pageNotFound: 'Page not found', serviceUnavailable: 'Service unavailable', unauthorized: 'Unauthorized' }, translatedErrorMessages);
    function getErrorIcon() {
        switch (errorType) {
            case '401': {
                return react_1.default.createElement(_401_1.default, { className: 'iui-non-ideal-state-illustration' });
            }
            case '403': {
                return react_1.default.createElement(_403_1.default, { className: 'iui-non-ideal-state-illustration' });
            }
            case '404': {
                return react_1.default.createElement(_404_1.default, { className: 'iui-non-ideal-state-illustration' });
            }
            case '500': {
                return react_1.default.createElement(_500_1.default, { className: 'iui-non-ideal-state-illustration' });
            }
            case '502': {
                return react_1.default.createElement(_502_1.default, { className: 'iui-non-ideal-state-illustration' });
            }
            case '503': {
                return react_1.default.createElement(_503_1.default, { className: 'iui-non-ideal-state-illustration' });
            }
            case 'generic':
            default: {
                return react_1.default.createElement(Error_1.default, { className: 'iui-non-ideal-state-illustration' });
            }
        }
    }
    function getHeadingMessage() {
        if (errorName) {
            return errorName;
        }
        switch (errorType) {
            case '401': {
                return defaultErrorMessages.unauthorized;
            }
            case '403': {
                return defaultErrorMessages.forbidden;
            }
            case '404': {
                return defaultErrorMessages.pageNotFound;
            }
            case '500': {
                return defaultErrorMessages.internalServerError;
            }
            case '502': {
                return defaultErrorMessages.badGateway;
            }
            case '503': {
                return defaultErrorMessages.serviceUnavailable;
            }
            case 'generic':
            default: {
                return defaultErrorMessages.error;
            }
        }
    }
    function getPrimaryButton() {
        if (!primaryButtonHandle || !primaryButtonLabel) {
            return undefined;
        }
        return (react_1.default.createElement(Button_1.Button, { styleType: 'high-visibility', onClick: primaryButtonHandle }, primaryButtonLabel));
    }
    function getSecondaryButton() {
        if (!secondaryButtonHandle || !secondaryButtonLabel) {
            return undefined;
        }
        return (react_1.default.createElement(Button_1.Button, { styleType: 'default', onClick: secondaryButtonHandle }, secondaryButtonLabel));
    }
    function getActions() {
        var primaryButton = getPrimaryButton();
        var secondaryButton = getSecondaryButton();
        if (!primaryButton && !secondaryButton) {
            return undefined;
        }
        return (react_1.default.createElement("div", { className: 'iui-non-ideal-state-actions' },
            primaryButton,
            secondaryButton));
    }
    return (react_1.default.createElement("div", __assign({ className: (0, classnames_1.default)('iui-non-ideal-state', className) }, rest),
        getErrorIcon(),
        react_1.default.createElement("div", { className: 'iui-non-ideal-state-title' }, getHeadingMessage()),
        errorMessage && (react_1.default.createElement("div", { className: 'iui-non-ideal-state-description' }, errorMessage)),
        getActions()));
};
exports.ErrorPage = ErrorPage;
exports.default = exports.ErrorPage;
