// @lifeomic/chroma-react/components/Alert/AlertBody.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var styles_1 = require("../../styles");
var clsx_1 = __importDefault(require("clsx"));
exports.AlertBodyStylesKey = 'ChromaAlertBody';
exports.useStyles = styles_1.makeStyles(function () { return ({
    root: {
        color: 'inherit',
        flex: '1 1 0%',
        '& > *': {
            color: 'inherit',
        },
        '& button': {
            color: 'inherit',
        },
        '& a': {
            color: 'inherit',
        },
        '& > p': {
            color: 'inherit',
        },
        '& > ul': {
            marginTop: '0.25rem',
            marginBottom: '0.25rem',
            paddingLeft: '1.25rem',
            paddingRight: '1.25rem',
        },
        '& p + a, & p + button, & ul + a, & ul + button': {
            marginLeft: '0.75rem',
        },
    },
    spaceBetween: {
        justifyContent: 'space-between',
        display: 'flex',
        '& a, & button': {
            minHeight: 'unset',
            padding: 0,
        },
        '@media screen and (max-width: 480px)': {
            alignItems: 'flex-start',
            '& button': {
                textAlign: 'left',
                justifyContent: 'flex-start',
                marginTop: '0.125rem',
            },
        },
    },
}); }, { name: exports.AlertBodyStylesKey });
exports.AlertBody = function (_a) {
    var className = _a.className, children = _a.children, spaceBetween = _a.spaceBetween, rootProps = __rest(_a, ["className", "children", "spaceBetween"]);
    var classes = exports.useStyles({});
    return (React.createElement("div", __assign({ className: clsx_1.default(classes.root, spaceBetween && classes.spaceBetween, className) }, rootProps), children));
};
