// @lifeomic/chroma-react/components/SecondaryNavigation/SecondaryNavigationItem.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecondaryNavigationItem = exports.useStyles = exports.SecondaryNavigationItemStylesKey = void 0;
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
var react_router_dom_1 = require("react-router-dom");
var styles_1 = require("../../styles");
exports.SecondaryNavigationItemStylesKey = 'ChromaSecondaryNavigationItem';
exports.useStyles = (0, styles_1.makeStyles)(function (theme) { return ({
    root: {
        paddingTop: theme.spacing(1.5),
        paddingBottom: theme.spacing(1.5),
        paddingRight: theme.spacing(3),
        textAlign: 'right',
        position: 'relative',
    },
    link: {
        color: theme.palette.text.hint,
        display: 'block',
        textDecoration: 'none',
        transition: 'color 0.5s ease',
        '&:hover': {
            textDecoration: 'none',
            color: theme.palette.primary.main,
        },
    },
    linkActive: {
        color: theme.palette.text.primary,
    },
    selector: {
        position: 'absolute',
        width: theme.pxToRem(4),
        height: '100%',
        top: 0,
        right: 0,
        background: theme.palette.primary.main,
        transition: 'all 0.2s',
        opacity: 0,
    },
    selectorActive: {
        opacity: 100,
    },
}); }, { name: exports.SecondaryNavigationItemStylesKey });
exports.SecondaryNavigationItem = React.forwardRef(function (_a, ref) {
    var className = _a.className, exact = _a.exact, label = _a.label, to = _a.to, additionalClasses = _a.classes, rootProps = __rest(_a, ["className", "exact", "label", "to", "classes"]);
    var classes = (0, exports.useStyles)({ classes: additionalClasses });
    return (React.createElement(react_router_dom_1.Route, { exact: exact, path: to }, function (_a) {
        var match = _a.match;
        var isActive = Boolean(match);
        return (React.createElement("li", __assign({ ref: ref, className: (0, clsx_1.default)(classes.root, className) }, rootProps),
            React.createElement(react_router_dom_1.NavLink, { activeClassName: classes.linkActive, className: classes.link, exact: exact, to: to }, label),
            React.createElement("div", { role: "presentation", className: (0, clsx_1.default)(classes.selector, isActive && classes.selectorActive) })));
    }));
});
