// @lifeomic/chroma-react/components/SecondaryNavigation/SecondaryNavigation.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecondaryNavigation = exports.useStyles = exports.SecondaryNavigationStylesKey = void 0;
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
var styles_1 = require("../../styles");
exports.SecondaryNavigationStylesKey = 'ChromaSecondaryNavigation';
exports.useStyles = (0, styles_1.makeStyles)(function (theme) { return ({
    root: {
        borderRight: "1px solid ".concat(theme.palette.divider),
        display: 'flex',
        flex: 1,
        flexDirection: 'column',
        height: '100%',
    },
    horizontalNav: {
        position: 'relative',
        borderRight: 0,
        '& > ul': {
            display: 'flex',
            borderBottom: "1px solid",
            borderBottomColor: theme.palette.divider,
            '& > li': {
                padding: theme.spacing(1, 3),
                fontSize: theme.typography.button.fontSize,
                letterSpacing: 0.5,
                whiteSpace: 'nowrap',
                '& > div': {
                    height: theme.pxToRem(2),
                    width: '100%',
                    top: 'auto',
                    bottom: 0,
                    left: 0,
                },
            },
        },
    },
    list: {
        listStyle: 'none',
        margin: 0,
        overflowY: 'auto',
        padding: 0,
    },
}); }, { name: exports.SecondaryNavigationStylesKey });
exports.SecondaryNavigation = React.forwardRef(function (_a, ref) {
    var children = _a.children, _b = _a.variant, variant = _b === void 0 ? 'vertical' : _b, className = _a.className, additionalClasses = _a.classes, rootProps = __rest(_a, ["children", "variant", "className", "classes"]);
    var classes = (0, exports.useStyles)({ classes: additionalClasses });
    return (React.createElement("nav", __assign({ ref: ref, className: (0, clsx_1.default)(classes.root, className, variant === 'horizontal' && classes.horizontalNav), "aria-label": "Secondary", role: "navigation" }, rootProps),
        React.createElement("ul", { className: classes.list }, children)));
});
