// @lifeomic/chroma-react/components/_private/forms/FormHelpMessage.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
var styles_1 = require("../../../styles");
var Text_1 = require("../../Text");
exports.FormHelpMessageStylesKey = 'ChromaFormHelpMessage';
var useStyles = styles_1.makeStyles(function (theme) { return ({
    root: {
        color: theme.palette.text.hint,
        lineHeight: 1.25,
        marginTop: theme.spacing(0.375),
    },
    inverse: {
        color: theme.palette.common.white,
        opacity: 0.9,
    },
}); }, { name: exports.FormHelpMessageStylesKey });
exports.FormHelpMessage = function (_a) {
    var className = _a.className, color = _a.color, children = _a.children, rootElementId = _a.rootElementId, describedById = _a.describedById, rootProps = __rest(_a, ["className", "color", "children", "rootElementId", "describedById"]);
    var classes = useStyles({});
    return (React.createElement(Text_1.Text, __assign({ size: "caption", "help-for": rootElementId, id: describedById, className: clsx_1.default(classes.root, color === 'inverse' && classes.inverse, className) }, rootProps), children));
};
