// @lifeomic/chroma-react/components/Tooltip/Tooltip.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tooltip = exports.useStyles = exports.TooltipStylesKey = void 0;
var createTypography_1 = require("../../styles/createTypography");
var styles_1 = require("../../styles");
var framer_motion_1 = require("framer-motion");
var Portal_1 = require("reakit/Portal");
var Tooltip_1 = require("reakit/Tooltip");
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
exports.TooltipStylesKey = 'ChromaTooltip';
exports.useStyles = (0, styles_1.makeStyles)(function (theme) { return ({
    root: {
        backgroundColor: theme.palette.common.black,
        borderRadius: theme.pxToRem(4),
        boxShadow: theme.boxShadows.tooltip,
        color: theme.palette.common.white,
        fontFamily: theme.typography.fontFamily,
        fontSize: createTypography_1.fontSizes.tooltip,
        letterSpacing: '0.021875em',
        maxWidth: theme.pxToRem(500),
        paddingLeft: theme.spacing(1),
        paddingRight: theme.spacing(1),
        paddingTop: theme.spacing(0.5),
        paddingBottom: theme.spacing(0.75),
        zIndex: 1300,
    },
}); }, { name: exports.TooltipStylesKey });
var Tooltip = function (_a) {
    var children = _a.children, className = _a.className, _b = _a.defaultVisible, defaultVisible = _b === void 0 ? false : _b, _c = _a.placement, placement = _c === void 0 ? 'bottom' : _c, _d = _a.title, title = _d === void 0 ? '' : _d, gutter = _a.gutter, rootProps = __rest(_a, ["children", "className", "defaultVisible", "placement", "title", "gutter"]);
    var classes = (0, exports.useStyles)({});
    var shouldReduceMotion = (0, framer_motion_1.useReducedMotion)();
    var tooltip = (0, Tooltip_1.useTooltipState)({
        placement: placement,
        visible: defaultVisible,
        gutter: gutter,
    });
    React.useEffect(function () {
        tooltip.place(placement);
    }, [placement, tooltip]);
    return (React.createElement(React.Fragment, null,
        React.createElement(Tooltip_1.TooltipReference, __assign({}, tooltip), function (referenceProps) {
            return React.cloneElement(React.Children.only(children), referenceProps);
        }),
        React.createElement(Portal_1.Portal, null,
            React.createElement(Tooltip_1.Tooltip, __assign({}, tooltip, { as: framer_motion_1.motion.div, className: (0, clsx_1.default)(classes.root, className), animate: tooltip.visible
                    ? shouldReduceMotion
                        ? { opacity: 1 }
                        : {
                            opacity: 1,
                            transition: { delay: 0.75 },
                        }
                    : {
                        opacity: 0,
                    } }, rootProps), title))));
};
exports.Tooltip = Tooltip;
