// @lifeomic/chroma-react/components/Stepper/StepConnector.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StepConnector = exports.useStyles = exports.StepConnectorStylesKey = void 0;
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
var styles_1 = require("../../styles");
exports.StepConnectorStylesKey = 'ChromaStepConnector';
exports.useStyles = (0, styles_1.makeStyles)(function (theme) { return ({
    connectorRoot: {
        flex: 1,
        position: 'relative',
        top: theme.pxToRem(-23),
        '@media screen and (max-width: 850px)': {
            display: 'none',
        },
    },
    subTitle: {
        top: theme.pxToRem(-33),
    },
    subTitlePill: {
        top: theme.pxToRem(-38),
    },
    line: {
        border: "1px solid ".concat(theme.palette.black[200]),
        display: 'block',
        position: 'absolute',
        right: theme.pxToRem(-16),
        transition: 'border .5s ease',
        width: "calc(100% + ".concat(theme.pxToRem(32), ")"),
    },
    activeLine: {
        border: "1px solid ".concat(theme.palette.primary.main),
    },
}); }, { name: exports.StepConnectorStylesKey });
var StepConnector = function (_a) {
    var active = _a.active, className = _a.className, completed = _a.completed, hasSubTitle = _a.hasSubTitle, hasSubTitlePill = _a.hasSubTitlePill, rootProps = __rest(_a, ["active", "className", "completed", "hasSubTitle", "hasSubTitlePill"]);
    var classes = (0, exports.useStyles)({});
    return (React.createElement("div", __assign({ className: (0, clsx_1.default)(classes.connectorRoot, hasSubTitle && classes.subTitle, hasSubTitlePill && classes.subTitlePill, className), role: "presentation" }, rootProps),
        React.createElement("span", { className: (0, clsx_1.default)(classes.line, (active || completed) && classes.activeLine) })));
};
exports.StepConnector = StepConnector;
