// @lifeomic/chroma-react/components/Stepper/Stepper.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
var styles_1 = require("../../styles");
var Box_1 = require("../Box");
var StepConnector_1 = require("./StepConnector");
exports.StepperStylesKey = 'ChromaStepper';
exports.useStyles = styles_1.makeStyles(function (theme) { return ({
    root: {
        backgroundColor: theme.palette.common.white,
    },
    innerRoot: {
        margin: theme.spacing(4, 4, 0, 4),
    },
}); }, { name: exports.StepperStylesKey });
exports.Stepper = function (_a) {
    var activeStep = _a.activeStep, _b = _a.as, as = _b === void 0 ? 'button' : _b, children = _a.children, className = _a.className, connectorClassName = _a.connectorClassName, onClick = _a.onClick, rootProps = __rest(_a, ["activeStep", "as", "children", "className", "connectorClassName", "onClick"]);
    var classes = exports.useStyles({});
    var childrenArray = Array.isArray(children)
        ? children
        : React.Children.toArray(children);
    var steps = childrenArray.map(function (child, index) {
        var childrenProps = {
            active: false,
            as: as,
            completed: false,
            index: index,
            onClick: onClick,
        };
        if (activeStep === index) {
            childrenProps.active = true;
        }
        else if (activeStep > index) {
            childrenProps.completed = true;
        }
        var connector = React.createElement(StepConnector_1.StepConnector, null);
        return [
            index !== 0 &&
                React.cloneElement(connector, __assign({ className: connectorClassName, hasSubTitle: child.props.subTitle, hasSubTitlePill: child.props.subTitlePillLabel, key: "connector-" + index }, childrenProps)),
            React.cloneElement(child, __assign({ key: "step-" + index }, childrenProps)),
        ];
    });
    return (React.createElement(Box_1.Box, __assign({ fullWidth: true, className: clsx_1.default(classes.root, className) }, rootProps),
        React.createElement(Box_1.Box, { align: "center", className: classes.innerRoot, fullWidth: true, justify: "space-between" }, steps)));
};
