// @lifeomic/chroma-react/components/Stepper/StepConnector.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
var styles_1 = require("../../styles");
exports.StepConnectorStylesKey = 'ChromaStepConnector';
exports.useStyles = styles_1.makeStyles(function (theme) { return ({
    connectorRoot: {
        flex: 1,
        position: 'relative',
        top: theme.pxToRem(-23),
        '@media screen and (max-width: 850px)': {
            display: 'none',
        },
    },
    subTitle: {
        top: theme.pxToRem(-33),
    },
    subTitlePill: {
        top: theme.pxToRem(-38),
    },
    line: {
        border: "1px solid " + theme.palette.black[200],
        display: 'block',
        position: 'absolute',
        right: theme.pxToRem(-16),
        transition: 'border .5s ease',
        width: "calc(100% + " + theme.pxToRem(32) + ")",
    },
    activeLine: {
        border: "1px solid " + theme.palette.primary.main,
    },
}); }, { name: exports.StepConnectorStylesKey });
exports.StepConnector = function (_a) {
    var active = _a.active, className = _a.className, completed = _a.completed, hasSubTitle = _a.hasSubTitle, hasSubTitlePill = _a.hasSubTitlePill, rootProps = __rest(_a, ["active", "className", "completed", "hasSubTitle", "hasSubTitlePill"]);
    var classes = exports.useStyles({});
    return (React.createElement("div", __assign({ className: clsx_1.default(classes.connectorRoot, hasSubTitle && classes.subTitle, hasSubTitlePill && classes.subTitlePill, className), role: "presentation" }, rootProps),
        React.createElement("span", { className: clsx_1.default(classes.line, (active || completed) && classes.activeLine) })));
};
