// @lifeomic/chroma-react/components/PrimaryNavigation/_private/common.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PLUS_EXIT_TRANSITION = exports.PLUS_ANIMATE_TRANSITION = exports.PLUS_INITIAL_TRANSITION = exports.BETA_EXIT_TRANSITION = exports.BETA_ANIMATE_TRANSITION = exports.BETA_INITIAL_TRANSITION = exports.delaySecondary = exports.outDurationSecondary = exports.inDurationSecondary = exports.outDuration = exports.inDuration = void 0;
exports.inDuration = 0.25;
exports.outDuration = 0.15;
exports.inDurationSecondary = 0.35;
exports.outDurationSecondary = 0.35;
exports.delaySecondary = 0.225;
exports.BETA_INITIAL_TRANSITION = {
    opacity: 0,
};
exports.BETA_ANIMATE_TRANSITION = {
    opacity: 1,
    transition: {
        duration: exports.inDurationSecondary,
        delay: exports.delaySecondary,
    },
};
exports.BETA_EXIT_TRANSITION = {
    opacity: 1,
    transition: {
        duration: exports.outDurationSecondary,
        delay: exports.delaySecondary,
    },
};
exports.PLUS_INITIAL_TRANSITION = {
    opacity: 0,
};
exports.PLUS_ANIMATE_TRANSITION = {
    opacity: 1,
    transition: {
        duration: exports.inDurationSecondary,
        delay: exports.delaySecondary,
    },
};
exports.PLUS_EXIT_TRANSITION = {
    opacity: 0,
    transition: {
        duration: exports.outDurationSecondary,
        delay: exports.delaySecondary,
    },
};
