// @lifeomic/chroma-react/esm/components/TableModule/TableModuleActions.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import * as React from 'react';
import { makeStyles } from '../../styles/index';
import clsx from 'clsx';
export var TableModuleActionsStylesKey = 'ChromaTableModuleActions';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        display: 'flex',
        justifyContent: 'flex-end',
        // Handle the case where an IconButton or IconbuttonLink is in a cell
        // We don't want it to be too large to push out the height of the row!
        '& > button, & > a': {
            display: 'flex',
            height: theme.pxToRem(26),
            marginRight: theme.spacing(2.5),
            padding: 0,
            '& > svg': {
                height: theme.pxToRem(15),
                width: theme.pxToRem(15),
            },
            '&:last-child': {
                marginRight: 0,
            },
        },
    },
}); }, { name: TableModuleActionsStylesKey });
export var TableModuleActions = function (_a) {
    var children = _a.children, className = _a.className, rootProps = __rest(_a, ["children", "className"]);
    var classes = useStyles({});
    return (React.createElement("div", __assign({ className: clsx(classes.root, className), "aria-label": "Table Actions" }, rootProps), children));
};
