// @lifeomic/chroma-react/esm/components/TableModule/TableActionDivider.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { Divider } from '../Divider';
import { makeStyles } from '../../styles';
import * as React from 'react';
import clsx from 'clsx';
export var TableActionDividerStylesKey = 'ChromaTableActionDivider';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        height: theme.pxToRem(19),
        margin: "0 ".concat(theme.pxToRem(16), " 0 0"),
        top: theme.pxToRem(4),
        position: 'relative',
    },
}); }, { name: TableActionDividerStylesKey });
export var TableActionDivider = function (_a) {
    var className = _a.className, rootProps = __rest(_a, ["className"]);
    var classes = useStyles({});
    return (React.createElement(Divider, __assign({ className: clsx(classes.root, className), color: "inverse", direction: "row" }, rootProps)));
};
