// @lifeomic/chroma-react/components/Select/GroupHeading.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
var styles_1 = require("../../styles");
exports.GroupHeadingStylesKey = 'ChromaSelectGroupHeading';
exports.useStyles = styles_1.makeStyles(function (theme) { return ({
    root: {
        fontSize: theme.pxToRem(10),
        fontWeight: theme.typography.fontWeightBold,
        letterSpacing: theme.pxToRem(1),
        padding: theme.spacing(0.25, 2, 0.5),
        color: theme.palette.text.hint,
        textTransform: 'uppercase',
        '&:not(:first-child)': {
            marginTop: theme.spacing(0.75),
            paddingTop: theme.spacing(1.25),
            borderTop: "solid 1px " + theme.palette.divider,
        },
    },
}); }, { name: exports.GroupHeadingStylesKey });
exports.GroupHeading = function (_a) {
    var children = _a.children, className = _a.className, dataSelectRole = _a["data-select-role"], rootProps = __rest(_a, ["children", "className", 'data-select-role']);
    var classes = exports.useStyles({});
    return (React.createElement("li", __assign({ className: clsx_1.default(classes.root, className), role: "option" }, rootProps), children));
};
