// @lifeomic/chroma-react/components/Select/RoverOption.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var Rover_1 = require("reakit/Rover");
var styles_1 = require("../../styles");
var framer_motion_1 = require("framer-motion");
exports.RoverOptionStylesKey = 'ChromaRoverOption';
exports.useStyles = styles_1.makeStyles(function (theme) { return ({
    option: {
        alignItems: 'center',
        backgroundColor: theme.palette.common.white,
        cursor: 'pointer',
        display: 'flex',
        minHeight: theme.pxToRem(26),
        outline: 'none',
        paddingLeft: theme.spacing(2),
        paddingRight: theme.spacing(2),
        transition: 'background-color 0.25s ease',
        '&:hover, &:focus': {
            backgroundColor: 'rgba(222,244,252, 0.6)',
        },
    },
}); }, { name: exports.RoverOptionStylesKey });
exports.RoverOption = React.memo(function (_a) {
    var handleSelectOption = _a.handleOptionSelect, option = _a.option, rover = _a.rover, variants = _a.variants, value = _a.value, disabled = _a.disabled;
    var _b;
    var classes = exports.useStyles({});
    var _c = (option === null || option === void 0 ? void 0 : option.props) || {}, optionValue = _c.value, meta = _c.meta;
    var handleClick = React.useCallback(function () {
        handleSelectOption(optionValue, meta);
    }, [optionValue, meta, handleSelectOption]);
    return (React.createElement(Rover_1.Rover, __assign({}, rover, { "aria-selected": optionValue === value, className: classes.option, as: framer_motion_1.motion.li, role: "option", value: (_b = option === null || option === void 0 ? void 0 : option.props) === null || _b === void 0 ? void 0 : _b.value, variants: variants, onClick: handleClick, disabled: disabled }), React.cloneElement(option, __assign({ isChecked: optionValue === value, disabled: disabled }, option.props))));
});
