// @lifeomic/chroma-react/components/Select/SelectOption.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
var VisuallyHidden_1 = require("reakit/VisuallyHidden");
var styles_1 = require("../../styles");
var Box_1 = require("../Box");
var Text_1 = require("../Text");
exports.SelectOptionStylesKey = 'ChromaSelectOption';
exports.checkSize = 20;
exports.useStyles = styles_1.makeStyles(function (theme) { return ({
    root: {
        paddingTop: theme.spacing(0.5),
        paddingBottom: theme.spacing(0.5),
    },
    title: {
        lineHeight: 1.3,
    },
    subtitle: {
        color: theme.palette.text.hint,
        lineHeight: 1.25,
        marginBottom: theme.spacing(0.25),
        marginTop: theme.spacing(0.25),
    },
    checked: {
        position: 'relative',
        '&::before': {
            background: theme.palette.primary.main,
            content: "''",
            display: 'block',
            height: "calc(100% + " + theme.spacing(1) + "px)",
            position: 'absolute',
            left: theme.spacing(-2),
            top: theme.spacing(-0.5),
            width: 4,
        },
        '& *': {
            color: theme.palette.primary.main,
            '&$title': {
                fontWeight: 900,
            },
        },
    },
    disabled: {
        cursor: 'not-allowed',
        opacity: 0.625,
    },
}); }, { name: exports.SelectOptionStylesKey });
exports.SelectOption = function (_a) {
    var className = _a.className, children = _a.children, isChecked = _a.isChecked, title = _a.title, subtitle = _a.subtitle, disabled = _a.disabled, rootProps = __rest(_a, ["className", "children", "isChecked", "title", "subtitle", "disabled"]);
    var classes = exports.useStyles({});
    return (React.createElement(Box_1.Box, __assign({ className: clsx_1.default(classes.root, className), fullWidth: true }, rootProps),
        isChecked && (
        // This is required for a11y: we need to have some indicator for our screenreader friends
        React.createElement(VisuallyHidden_1.VisuallyHidden, null, "\u2713")),
        title && (React.createElement(Box_1.Box, { direction: "column", justify: "flex-start", className: clsx_1.default(isChecked && classes.checked, disabled && classes.disabled) },
            title && (React.createElement(Text_1.Text, { className: classes.title, size: "subbody" }, title)),
            subtitle && (React.createElement(Text_1.Text, { className: classes.subtitle, size: "caption" }, subtitle)))),
        children));
};
