// @lifeomic/chroma-react/esm/components/Breadcrumbs/Breadcrumb.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { Link } from 'react-router-dom';
import { makeStyles } from '../../styles/index';
export var BreadcrumbStylesKey = 'ChromaBreadcrumb';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        display: 'inline',
        margin: 0,
        padding: 0,
        '& + li:before': {
            display: 'inline-block',
            marginTop: 0,
            marginBottom: 0,
            marginLeft: theme.spacing(1),
            marginRight: theme.spacing(1),
            transform: 'rotate(15deg)',
            borderRight: "".concat(theme.pxToRem(1), " solid ").concat(theme.palette.text.dark),
            height: '0.75rem',
            content: "''",
        },
    },
    rootInverse: {
        '& + li:before': {
            borderRight: "".concat(theme.pxToRem(1), " solid ").concat(theme.palette.common.white),
        },
    },
    link: {
        fontSize: theme.pxToRem(16),
        letterSpacing: theme.pxToRem(0.5),
        textDecoration: 'none',
        color: theme.palette.text.dark,
        transition: 'color 0.5s ease',
        '&:hover': {
            color: theme.palette.primary.main,
            textDecoration: 'underline',
        },
    },
    linkInverse: {
        color: theme.palette.common.white,
        transition: 'opacity 0.5s ease',
        opacity: 1,
        '&:hover': {
            color: theme.palette.common.white,
            opacity: 0.85,
        },
    },
}); }, { name: BreadcrumbStylesKey });
export var Breadcrumb = function (_a) {
    var _b, _c;
    var className = _a.className, _d = _a.color, color = _d === void 0 ? 'default' : _d, _e = _a.isCurrentPage, isCurrentPage = _e === void 0 ? false : _e, text = _a.text, url = _a.url, rootProps = __rest(_a, ["className", "color", "isCurrentPage", "text", "url"]);
    var classes = useStyles({});
    return (React.createElement("li", __assign({ className: clsx(classes.root, (_b = {},
            _b[classes.rootInverse] = color === 'inverse',
            _b), className) }, rootProps),
        React.createElement(Link, { "aria-current": isCurrentPage ? 'page' : undefined, className: clsx(classes.link, (_c = {},
                _c[classes.linkInverse] = color === 'inverse',
                _c)), to: url }, text)));
};
