// @lifeomic/chroma-react/esm/components/Breadcrumbs/BreadcrumbNav.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { makeStyles } from '../../styles/index';
export var BreadcrumbNavStylesKey = 'ChromaBreadcrumbNav';
export var useStyles = makeStyles(function (theme) { return ({
    root: {},
    ol: {
        margin: theme.spacing(0),
        paddingLeft: theme.spacing(0),
        listStyle: 'none',
    },
}); }, { name: BreadcrumbNavStylesKey });
export var BreadcrumbNav = function (_a) {
    var className = _a.className, children = _a.children, rootProps = __rest(_a, ["className", "children"]);
    var classes = useStyles({});
    return (React.createElement("nav", __assign({ "aria-label": "Breadcrumb", className: clsx(classes.root, className) }, rootProps),
        React.createElement("ol", { className: classes.ol }, children)));
};
