// @lifeomic/chroma-react/esm/components/SecondaryNavigation/SecondaryNavigation.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { makeStyles } from '../../styles';
export var SecondaryNavigationStylesKey = 'ChromaSecondaryNavigation';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        borderRight: "1px solid ".concat(theme.palette.divider),
        display: 'flex',
        flex: 1,
        flexDirection: 'column',
        height: '100%',
    },
    horizontalNav: {
        position: 'relative',
        borderRight: 0,
        '& > ul': {
            display: 'flex',
            borderBottom: "1px solid",
            borderBottomColor: theme.palette.divider,
            '& > li': {
                padding: theme.spacing(1, 3),
                fontSize: theme.typography.button.fontSize,
                letterSpacing: 0.5,
                whiteSpace: 'nowrap',
                '& > div': {
                    height: theme.pxToRem(2),
                    width: '100%',
                    top: 'auto',
                    bottom: 0,
                    left: 0,
                },
            },
        },
    },
    list: {
        listStyle: 'none',
        margin: 0,
        overflowY: 'auto',
        padding: 0,
    },
}); }, { name: SecondaryNavigationStylesKey });
export var SecondaryNavigation = React.forwardRef(function (_a, ref) {
    var children = _a.children, _b = _a.variant, variant = _b === void 0 ? 'vertical' : _b, className = _a.className, additionalClasses = _a.classes, rootProps = __rest(_a, ["children", "variant", "className", "classes"]);
    var classes = useStyles({ classes: additionalClasses });
    return (React.createElement("nav", __assign({ ref: ref, className: clsx(classes.root, className, variant === 'horizontal' && classes.horizontalNav), "aria-label": "Secondary", role: "navigation" }, rootProps),
        React.createElement("ul", { className: classes.list }, children)));
});
