// @lifeomic/chroma-react/esm/components/SecondaryNavigation/SecondaryNavigationItem.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { NavLink, Route } from 'react-router-dom';
import { makeStyles } from '../../styles';
export var SecondaryNavigationItemStylesKey = 'ChromaSecondaryNavigationItem';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        paddingTop: theme.spacing(1.5),
        paddingBottom: theme.spacing(1.5),
        paddingRight: theme.spacing(3),
        textAlign: 'right',
        position: 'relative',
    },
    link: {
        color: theme.palette.text.hint,
        display: 'block',
        textDecoration: 'none',
        transition: 'color 0.5s ease',
        '&:hover': {
            textDecoration: 'none',
            color: theme.palette.primary.main,
        },
    },
    linkActive: {
        color: theme.palette.text.primary,
    },
    selector: {
        position: 'absolute',
        width: theme.pxToRem(4),
        height: '100%',
        top: 0,
        right: 0,
        background: theme.palette.primary.main,
        transition: 'all 0.2s',
        opacity: 0,
    },
    selectorActive: {
        opacity: 100,
    },
}); }, { name: SecondaryNavigationItemStylesKey });
export var SecondaryNavigationItem = React.forwardRef(function (_a, ref) {
    var className = _a.className, exact = _a.exact, label = _a.label, to = _a.to, additionalClasses = _a.classes, rootProps = __rest(_a, ["className", "exact", "label", "to", "classes"]);
    var classes = useStyles({ classes: additionalClasses });
    return (React.createElement(Route, { exact: exact, path: to }, function (_a) {
        var match = _a.match;
        var isActive = Boolean(match);
        return (React.createElement("li", __assign({ ref: ref, className: clsx(classes.root, className) }, rootProps),
            React.createElement(NavLink, { activeClassName: classes.linkActive, className: classes.link, exact: exact, to: to }, label),
            React.createElement("div", { role: "presentation", className: clsx(classes.selector, isActive && classes.selectorActive) })));
    }));
});
