// @lifeomic/chroma-react/esm/components/_private/forms/FormErrorMessage.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { makeStyles } from '../../../styles';
import { Text } from '../../Text';
export var FormErrorMessageStylesKey = 'ChromaFormErrorMessage';
var useStyles = makeStyles(function (theme) { return ({
    root: {
        color: theme.palette.error.main,
        lineHeight: 1.25,
        marginTop: theme.spacing(0.375),
    },
    inverse: {
        color: theme.palette.error[50],
    },
}); }, { name: FormErrorMessageStylesKey });
export var FormErrorMessage = function (_a) {
    var className = _a.className, describedById = _a.describedById, color = _a.color, children = _a.children, rootElementId = _a.rootElementId, rootProps = __rest(_a, ["className", "describedById", "color", "children", "rootElementId"]);
    var classes = useStyles({});
    return (React.createElement(Text, __assign({ size: "caption", weight: "bold", "error-for": rootElementId, id: describedById, "aria-live": "assertive", role: "alert", className: clsx(classes.root, color === 'inverse' && classes.inverse, className) }, rootProps), children));
};
