// @itwin/itwinui-react/cjs/core/utils/common.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getBoundedValue = exports.getWindow = exports.getDocument = exports.getContainer = exports.getUserColor = exports.StatusIconMap = void 0;
/*---------------------------------------------------------------------------------------------
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
var InfoCircular_1 = __importDefault(require("@itwin/itwinui-icons-react/cjs/icons/InfoCircular"));
var StatusError_1 = __importDefault(require("@itwin/itwinui-icons-react/cjs/icons/StatusError"));
var StatusSuccess_1 = __importDefault(require("@itwin/itwinui-icons-react/cjs/icons/StatusSuccess"));
var StatusWarning_1 = __importDefault(require("@itwin/itwinui-icons-react/cjs/icons/StatusWarning"));
var react_1 = __importDefault(require("react"));
exports.StatusIconMap = {
    negative: function (args) { return react_1.default.createElement(StatusError_1.default, __assign({ "aria-hidden": true }, args)); },
    positive: function (args) { return react_1.default.createElement(StatusSuccess_1.default, __assign({ "aria-hidden": true }, args)); },
    warning: function (args) { return react_1.default.createElement(StatusWarning_1.default, __assign({ "aria-hidden": true }, args)); },
    informational: function (args) { return (react_1.default.createElement(InfoCircular_1.default, __assign({ "aria-hidden": true }, args))); },
};
var USER_COLORS = [
    '#6AB9EC',
    '#B1C854',
    '#F7706C',
    '#4585A5',
    '#FFC335',
    '#F7963E',
    '#73C7C1',
    '#85A9CF',
    '#A3779F',
    '#C8C2B4',
    '#A47854',
];
/**
 * Generate color from user name or email.
 * Recommended to use for `backgroundColor` in `UserIcon` component.
 */
var getUserColor = function (emailOrName) {
    var normalizedString = emailOrName.trim().toLowerCase();
    var hash = 0;
    for (var i = 0; i < normalizedString.length; i++) {
        var charCode = normalizedString.charCodeAt(i);
        hash = (hash + charCode) % USER_COLORS.length;
    }
    return USER_COLORS[hash];
};
exports.getUserColor = getUserColor;
/**
 * Get the container as a child of body, or create one if it doesn't exist.
 * Mostly used for dynamic components like Modal or Toast.
 *
 * @param containerId id of the container to find or create
 * @param ownerDocument Can be changed if the container should be in a different document (e.g. in popup).
 */
var getContainer = function (containerId, ownerDocument) {
    var _a;
    if (ownerDocument === void 0) { ownerDocument = exports.getDocument(); }
    var container = (_a = ownerDocument === null || ownerDocument === void 0 ? void 0 : ownerDocument.getElementById(containerId)) !== null && _a !== void 0 ? _a : undefined;
    if (container == null && !!ownerDocument) {
        container = ownerDocument.createElement('div');
        container.setAttribute('id', containerId);
        ownerDocument.body.appendChild(container);
    }
    return container;
};
exports.getContainer = getContainer;
/**
 * Get document if it is defined.
 * Used to support SSR/SSG applications.
 */
var getDocument = function () {
    return typeof document === 'undefined' ? undefined : document;
};
exports.getDocument = getDocument;
/**
 * Get window if it is defined.
 * Used to support SSR/SSG applications.
 */
var getWindow = function () {
    return typeof window === 'undefined' ? undefined : window;
};
exports.getWindow = getWindow;
/**
 * Return input value bounded by specified range.
 */
var getBoundedValue = function (val, min, max) {
    return Math.min(max, Math.max(min, val));
};
exports.getBoundedValue = getBoundedValue;
