// @lifeomic/chroma-react/styles/overrides/MuiButton.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createMuiButtonOverrides = void 0;
var createMuiButtonOverrides = function (theme) {
    var basePadding = "".concat(theme.spacing(1.375), "px ").concat(theme.spacing(1.75), "px");
    return {
        root: __assign(__assign({}, theme.typography.button), { textTransform: 'initial', boxSizing: 'border-box', minWidth: theme.spacing(11), minHeight: theme.pxToRem(30), padding: basePadding, transition: theme.transitions.create(['background-color', 'color', 'box-shadow', 'border'], {
                duration: theme.transitions.duration.short,
            }), '&:focus': {
                outline: 0,
            } }),
        textPrimary: {
            // text, contained, and outlined buttons should all be the same size
            padding: basePadding,
            color: theme.palette.primary.main,
            '&:hover, &:focus': {
                backgroundColor: 'inherit',
                color: theme.palette.primary[800],
            },
            '&:disabled': {
                color: theme.palette.primary.main,
                opacity: 0.44,
            },
        },
        outlinedPrimary: {
            border: "1px solid ".concat(theme.palette.primary.main),
            '&:hover, &:focus': {
                backgroundColor: 'transparent',
                color: theme.palette.primary[800],
                border: "1px solid ".concat(theme.palette.primary[800]),
            },
            '&:disabled': {
                color: theme.palette.primary.main,
                opacity: 0.44,
                border: "1px solid ".concat(theme.palette.primary.main),
            },
        },
        containedPrimary: {
            color: theme.palette.common.white,
            backgroundColor: theme.palette.primary.main,
            boxShadow: 'none',
            '&$focusVisible': {
                boxShadow: theme.shadows[6],
            },
            '&:disabled': {
                color: theme.palette.common.white,
                backgroundColor: theme.palette.primary.main,
                opacity: 0.44,
            },
            '&:active': {
                // Remove boxshadow "pop" when clicking
                boxShadow: 'none',
            },
            '&:hover, &:focus': {
                backgroundColor: theme.palette.primary[800],
                // Reset on touch devices, it doesn't add specificity
                '@media (hover: none)': {
                    backgroundColor: theme.palette.grey[300],
                },
            },
        },
        sizeSmall: __assign(__assign({}, theme.typography.button), { padding: "".concat(theme.spacing(1), "px ").concat(theme.spacing(1.375), "px"), minWidth: theme.spacing(8), minHeight: theme.pxToRem(32) }),
    };
};
exports.createMuiButtonOverrides = createMuiButtonOverrides;
