// @lifeomic/chroma-react/styles/overrides/MuiTab.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createMuiTabOverrides = void 0;
var createMuiTabOverrides = function (theme) {
    var _a;
    return ({
        root: (_a = {
                minHeight: theme.pxToRem(48),
                fontWeight: theme.typography.fontWeightRegular,
                fontSize: theme.typography.subtitle1.fontSize,
                textTransform: 'initial',
                transition: 'color 0.5s ease',
                letterSpacing: theme.pxToRem(0.1),
                '&:hover': {
                    color: theme.palette.black[500],
                },
                '&:disabled': {
                    color: theme.palette.black[400],
                },
                '&:focus': {
                    outline: 'none',
                },
                '&$selected': {
                    color: theme.palette.black[900],
                }
            },
            _a[theme.breakpoints.up('sm')] = {
                fontSize: theme.typography.subtitle1.fontSize,
            },
            _a),
        textColorInherit: {
            color: theme.palette.black[900],
            opacity: 'initial',
        },
        wrapper: {
            paddingLeft: theme.spacing(2),
            paddingRight: theme.spacing(2),
        },
    });
};
exports.createMuiTabOverrides = createMuiTabOverrides;
