// @lifeomic/chroma-react/esm/components/Paper/Paper.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { makeStyles } from '../../styles';
import { Box } from '../Box';
export var PaperStylesKey = 'ChromaPaper';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        background: theme.palette.common.white,
        boxShadow: theme.boxShadows.table,
        borderRadius: theme.pxToRem(3),
    },
    padding0: {
        padding: 0,
    },
    padding1: {
        padding: theme.spacing(1),
    },
    padding2: {
        padding: theme.spacing(2),
    },
}); }, { name: PaperStylesKey });
export var Paper = React.forwardRef(function (_a, ref) {
    var _b;
    var children = _a.children, className = _a.className, _c = _a.padding, padding = _c === void 0 ? 2 : _c, rootProps = __rest(_a, ["children", "className", "padding"]);
    var classes = useStyles({});
    return (React.createElement(Box, __assign({ direction: "column", ref: ref, className: clsx(classes.root, (_b = {},
            _b[classes.padding0] = padding === 0,
            _b[classes.padding1] = padding === 1,
            _b[classes.padding2] = padding === 2,
            _b), className) }, rootProps), children));
});
