// @lifeomic/chroma-react/esm/components/Stepper/Stepper.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { makeStyles } from '../../styles';
import { Box } from '../Box';
import { StepConnector } from './StepConnector';
export var StepperStylesKey = 'ChromaStepper';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        backgroundColor: theme.palette.common.white,
    },
    innerRoot: {
        margin: theme.spacing(4, 4, 0, 4),
    },
}); }, { name: StepperStylesKey });
export var Stepper = function (_a) {
    var activeStep = _a.activeStep, _b = _a.as, as = _b === void 0 ? 'button' : _b, children = _a.children, className = _a.className, connectorClassName = _a.connectorClassName, onClick = _a.onClick, rootProps = __rest(_a, ["activeStep", "as", "children", "className", "connectorClassName", "onClick"]);
    var classes = useStyles({});
    var childrenArray = Array.isArray(children)
        ? children
        : React.Children.toArray(children);
    var steps = childrenArray.map(function (child, index) {
        var childrenProps = {
            active: false,
            as: as,
            completed: false,
            index: index,
            onClick: onClick,
        };
        if (activeStep === index) {
            childrenProps.active = true;
        }
        else if (activeStep > index) {
            childrenProps.completed = true;
        }
        var connector = React.createElement(StepConnector, null);
        return [
            index !== 0 &&
                React.cloneElement(connector, __assign({ className: connectorClassName, hasSubTitle: child.props.subTitle, hasSubTitlePill: child.props.subTitlePillLabel, key: "connector-" + index }, childrenProps)),
            React.cloneElement(child, __assign({ key: "step-" + index }, childrenProps)),
        ];
    });
    return (React.createElement(Box, __assign({ fullWidth: true, className: clsx(classes.root, className) }, rootProps),
        React.createElement(Box, { align: "center", className: classes.innerRoot, fullWidth: true, justify: "space-between" }, steps)));
};
