// @lifeomic/chroma-react/esm/components/Select/GroupHeading.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { makeStyles } from '../../styles';
export var GroupHeadingStylesKey = 'ChromaSelectGroupHeading';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        fontSize: theme.pxToRem(10),
        fontWeight: theme.typography.fontWeightBold,
        letterSpacing: theme.pxToRem(1),
        padding: theme.spacing(0.25, 2, 0.5),
        color: theme.palette.text.hint,
        textTransform: 'uppercase',
        '&:not(:first-child)': {
            marginTop: theme.spacing(0.75),
            paddingTop: theme.spacing(1.25),
            borderTop: "solid 1px ".concat(theme.palette.divider),
        },
    },
}); }, { name: GroupHeadingStylesKey });
export var GroupHeading = function (_a) {
    var children = _a.children, className = _a.className, dataSelectRole = _a["data-select-role"], rootProps = __rest(_a, ["children", "className", 'data-select-role']);
    var classes = useStyles({});
    return (React.createElement("li", __assign({ className: clsx(classes.root, className), role: "option" }, rootProps), children));
};
