// @lifeomic/chroma-react/esm/components/Select/RoverOption.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import * as React from 'react';
import { Rover } from 'reakit/Rover';
import { makeStyles } from '../../styles';
import { motion } from 'framer-motion';
export var RoverOptionStylesKey = 'ChromaRoverOption';
export var useStyles = makeStyles(function (theme) { return ({
    option: {
        alignItems: 'center',
        backgroundColor: theme.palette.common.white,
        cursor: 'pointer',
        display: 'flex',
        minHeight: theme.pxToRem(26),
        outline: 'none',
        paddingLeft: theme.spacing(2),
        paddingRight: theme.spacing(2),
        transition: 'background-color 0.25s ease',
        '&:hover, &:focus': {
            backgroundColor: 'rgba(222,244,252, 0.6)',
        },
    },
}); }, { name: RoverOptionStylesKey });
export var RoverOption = React.memo(function (_a) {
    var handleSelectOption = _a.handleOptionSelect, option = _a.option, rover = _a.rover, variants = _a.variants, value = _a.value, disabled = _a.disabled;
    var _b;
    var classes = useStyles({});
    var _c = (option === null || option === void 0 ? void 0 : option.props) || {}, optionValue = _c.value, meta = _c.meta;
    var handleClick = React.useCallback(function () {
        handleSelectOption(optionValue, meta);
    }, [optionValue, meta, handleSelectOption]);
    return (React.createElement(Rover, __assign({}, rover, { "aria-selected": optionValue === value, className: classes.option, as: motion.li, role: "option", value: (_b = option === null || option === void 0 ? void 0 : option.props) === null || _b === void 0 ? void 0 : _b.value, variants: variants, onClick: handleClick, disabled: disabled }), React.cloneElement(option, __assign({ isChecked: optionValue === value, disabled: disabled }, option.props))));
});
