// @lifeomic/chroma-react/esm/components/Select/SelectOption.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { VisuallyHidden } from 'reakit/VisuallyHidden';
import { makeStyles } from '../../styles';
import { Box } from '../Box';
import { Text } from '../Text';
export var SelectOptionStylesKey = 'ChromaSelectOption';
export var checkSize = 20;
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        paddingTop: theme.spacing(0.5),
        paddingBottom: theme.spacing(0.5),
    },
    title: {
        lineHeight: 1.3,
    },
    subtitle: {
        color: theme.palette.text.hint,
        lineHeight: 1.25,
        marginBottom: theme.spacing(0.25),
        marginTop: theme.spacing(0.25),
    },
    checked: {
        position: 'relative',
        '&::before': {
            background: theme.palette.primary.main,
            content: "''",
            display: 'block',
            height: "calc(100% + ".concat(theme.spacing(1), "px)"),
            position: 'absolute',
            left: theme.spacing(-2),
            top: theme.spacing(-0.5),
            width: 4,
        },
        '& *': {
            color: theme.palette.primary.main,
            '&$title': {
                fontWeight: 900,
            },
        },
    },
    disabled: {
        cursor: 'not-allowed',
        opacity: 0.625,
    },
}); }, { name: SelectOptionStylesKey });
export var SelectOption = function (_a) {
    var className = _a.className, children = _a.children, isChecked = _a.isChecked, title = _a.title, subtitle = _a.subtitle, disabled = _a.disabled, rootProps = __rest(_a, ["className", "children", "isChecked", "title", "subtitle", "disabled"]);
    var classes = useStyles({});
    return (React.createElement(Box, __assign({ className: clsx(classes.root, className), fullWidth: true }, rootProps),
        isChecked && (
        // This is required for a11y: we need to have some indicator for our screenreader friends
        React.createElement(VisuallyHidden, null, "\u2713")),
        title && (React.createElement(Box, { direction: "column", justify: "flex-start", className: clsx(isChecked && classes.checked, disabled && classes.disabled) },
            title && (React.createElement(Text, { className: classes.title, size: "subbody" }, title)),
            subtitle && (React.createElement(Text, { className: classes.subtitle, size: "caption" }, subtitle)))),
        children));
};
