// @lifeomic/chroma-react/esm/components/Popover/Popover.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { ConditionalWrapper } from '../_private/ConditionalWrapper';
import { makeStyles } from '../../styles';
import { motion, useReducedMotion } from 'framer-motion';
import { Portal } from 'reakit/Portal';
import { Text } from '../Text';
import { Popover as ReakitPopover, PopoverDisclosure, usePopoverState, } from 'reakit/Popover';
import clsx from 'clsx';
import * as React from 'react';
export var PopoverStylesKey = 'ChromaPopover';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        background: theme.palette.common.white,
        color: theme.palette.text.primary,
        borderRadius: theme.pxToRem(10),
        boxShadow: theme.boxShadows.popover,
        minWidth: theme.pxToRem(224),
        overflow: 'hidden',
        zIndex: 40,
        '&:focus': {
            outline: 'none',
        },
    },
    title: {
        paddingTop: theme.spacing(2),
        paddingLeft: theme.spacing(2.5),
        paddingRight: theme.spacing(2.5),
    },
}); }, { name: PopoverStylesKey });
export var Popover = function (_a) {
    var ariaLabel = _a["aria-label"], anchorElement = _a.anchorElement, children = _a.children, className = _a.className, gutter = _a.gutter, _b = _a.placement, placement = _b === void 0 ? 'bottom' : _b, title = _a.title, _c = _a.usePortal, usePortal = _c === void 0 ? false : _c, unstable_offset = _a.unstable_offset, rootProps = __rest(_a, ['aria-label', "anchorElement", "children", "className", "gutter", "placement", "title", "usePortal", "unstable_offset"]);
    var classes = useStyles({});
    var shouldReduceMotion = useReducedMotion();
    var popover = usePopoverState({ placement: placement, gutter: gutter, unstable_offset: unstable_offset });
    return (React.createElement(React.Fragment, null,
        React.createElement(PopoverDisclosure, __assign({}, popover, anchorElement.props), function (disclosureProps) {
            return React.cloneElement(React.Children.only(anchorElement), disclosureProps);
        }),
        React.createElement(ConditionalWrapper, { condition: usePortal, wrapper: function (children) { return React.createElement(Portal, null, children); } },
            React.createElement(ReakitPopover, __assign({}, popover, { "aria-label": ariaLabel, as: motion.div, className: clsx(classes.root, className), "aria-hidden": popover.visible ? undefined : 'true', animate: popover.visible
                    ? shouldReduceMotion
                        ? { opacity: 1 }
                        : {
                            opacity: 1,
                            transition: { duration: 0.3 },
                        }
                    : shouldReduceMotion
                        ? { opacity: 0 }
                        : {
                            opacity: 0,
                            transition: { duration: 0.1 },
                        } }, rootProps),
                !!title && (React.createElement(Text, { className: classes.title, weight: "bold", size: "subbody" }, title)),
                typeof children === 'function'
                    ? children({ popover: popover })
                    : children))));
};
