// @lifeomic/chroma-react/esm/components/Popover/PopoverList.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import * as React from 'react';
import { makeStyles } from '../../styles';
export var PopoverListStylesKey = 'ChromaPopoverList';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        listStyle: 'none',
        padding: 0,
        margin: 0,
        paddingTop: theme.spacing(2),
        paddingBottom: theme.spacing(2),
        maxHeight: theme.pxToRem(432),
        overflowY: 'auto',
        '&:focus': {
            outline: 'none',
        },
    },
}); }, { name: PopoverListStylesKey });
export var PopoverList = React.forwardRef(function (_a, ref) {
    var children = _a.children, rootProps = __rest(_a, ["children"]);
    var classes = useStyles({});
    return (React.createElement("ul", __assign({ ref: ref, className: classes.root, tabIndex: 0 }, rootProps), children));
});
