// @lifeomic/chroma-react/esm/components/Modal/ModalActions.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { makeStyles } from '../../styles';
export var ModalActionsStylesKey = 'ChromaModalActions';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        display: 'flex',
        borderTop: "1px solid " + theme.palette.divider,
        paddingTop: theme.spacing(2),
        paddingBottom: theme.spacing(1.75),
        '& > button': {
            marginRight: theme.spacing(1),
            '&:last-of-type': {
                marginRight: 0,
            },
        },
    },
    justifyEnd: {
        justifyContent: 'flex-end',
    },
    justifyBetween: {
        justifyContent: 'space-between',
    },
}); }, { name: ModalActionsStylesKey });
export var ModalActions = function (_a) {
    var _b;
    var className = _a.className, children = _a.children, _c = _a.justify, justify = _c === void 0 ? 'flex-end' : _c, rootProps = __rest(_a, ["className", "children", "justify"]);
    var classes = useStyles({});
    return (React.createElement("div", __assign({ className: clsx(classes.root, (_b = {},
            _b[classes.justifyBetween] = justify === 'space-between',
            _b[classes.justifyEnd] = justify === 'flex-end',
            _b), className) }, rootProps), children));
};
