// @lifeomic/chroma-react/esm/components/ButtonUnstyled/ButtonUnstyled.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import * as React from 'react';
import { makeStyles } from '../../styles';
import clsx from 'clsx';
export var ButtonUnstyledStylesKey = 'ChromaButtonUnstyled';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        backgroundColor: 'transparent',
        border: 0,
        borderRadius: theme.pxToRem(4),
        color: 'inherit',
        cursor: 'pointer',
        padding: 0,
        width: 'fit-content',
        '&:focus': {
            outline: 'none',
        },
        '&:focus.focus-visible': {
            boxShadow: theme.boxShadows.focusVisible,
        },
        '&::-moz-focus-inner': {
            border: 'none',
        },
    },
    fullWidth: {
        width: '100%',
    },
}); }, { name: ButtonUnstyledStylesKey });
export var ButtonUnstyled = React.forwardRef(function (_a, ref) {
    var children = _a.children, className = _a.className, disabled = _a.disabled, fullWidth = _a.fullWidth, rootProps = __rest(_a, ["children", "className", "disabled", "fullWidth"]);
    var classes = useStyles({});
    return (React.createElement("button", __assign({ className: clsx(classes.root, fullWidth && classes.fullWidth, className), disabled: disabled, ref: ref, tabIndex: disabled ? -1 : 0, type: "button" }, rootProps), children));
});
