// @lifeomic/chroma-react/components/SlideOver/Header.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var IconButton_1 = require("../IconButton");
var styles_1 = require("../../styles");
var Text_1 = require("../Text");
var chromicons_1 = require("@lifeomic/chromicons");
var React = __importStar(require("react"));
var clsx_1 = __importDefault(require("clsx"));
exports.HeaderStylesKey = 'ChromaSlideOverHeader';
exports.useStyles = styles_1.makeStyles(function (theme) { return ({
    root: {
        alignItems: 'center',
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'space-between',
        backgroundColor: theme.palette.black[50],
        padding: theme.spacing(2),
    },
}); }, { name: exports.HeaderStylesKey });
exports.Header = function (_a) {
    var children = _a.children, additionalClasses = _a.classes, className = _a.className, onClose = _a.onClose, title = _a.title, rootProps = __rest(_a, ["children", "classes", "className", "onClose", "title"]);
    var classes = exports.useStyles({});
    return (React.createElement("div", __assign({ className: clsx_1.default(classes.root, additionalClasses === null || additionalClasses === void 0 ? void 0 : additionalClasses.root, className) }, rootProps),
        children,
        title && (React.createElement(Text_1.Text, { className: clsx_1.default(additionalClasses === null || additionalClasses === void 0 ? void 0 : additionalClasses.text), size: "body", weight: "bold" }, title)),
        React.createElement(IconButton_1.IconButton, { className: additionalClasses === null || additionalClasses === void 0 ? void 0 : additionalClasses.button, "aria-label": "Close panel", icon: chromicons_1.X, onClick: onClose, size: 0 })));
};
