// @itwin/itwinui-react/cjs/core/Typography/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Text = exports.KbdKeys = exports.Kbd = exports.Code = exports.Blockquote = exports.Small = exports.Body = exports.Leading = exports.Subheading = exports.Title = exports.Headline = exports.Anchor = void 0;
/*---------------------------------------------------------------------------------------------
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
var Anchor_1 = require("./Anchor");
Object.defineProperty(exports, "Anchor", { enumerable: true, get: function () { return Anchor_1.Anchor; } });
var Headline_1 = require("./Headline");
Object.defineProperty(exports, "Headline", { enumerable: true, get: function () { return Headline_1.Headline; } });
var Title_1 = require("./Title");
Object.defineProperty(exports, "Title", { enumerable: true, get: function () { return Title_1.Title; } });
var Subheading_1 = require("./Subheading");
Object.defineProperty(exports, "Subheading", { enumerable: true, get: function () { return Subheading_1.Subheading; } });
var Leading_1 = require("./Leading");
Object.defineProperty(exports, "Leading", { enumerable: true, get: function () { return Leading_1.Leading; } });
var Body_1 = require("./Body");
Object.defineProperty(exports, "Body", { enumerable: true, get: function () { return Body_1.Body; } });
var Small_1 = require("./Small");
Object.defineProperty(exports, "Small", { enumerable: true, get: function () { return Small_1.Small; } });
var Blockquote_1 = require("./Blockquote");
Object.defineProperty(exports, "Blockquote", { enumerable: true, get: function () { return Blockquote_1.Blockquote; } });
var Code_1 = require("./Code");
Object.defineProperty(exports, "Code", { enumerable: true, get: function () { return Code_1.Code; } });
var Kbd_1 = require("./Kbd");
Object.defineProperty(exports, "Kbd", { enumerable: true, get: function () { return Kbd_1.Kbd; } });
Object.defineProperty(exports, "KbdKeys", { enumerable: true, get: function () { return Kbd_1.KbdKeys; } });
var Text_1 = require("./Text");
Object.defineProperty(exports, "Text", { enumerable: true, get: function () { return Text_1.Text; } });
