// @lifeomic/chroma-react/esm/components/Alert/AlertActionsRow.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import * as React from 'react';
import { makeStyles } from '../../styles';
import clsx from 'clsx';
export var AlertActionsRowStylesKey = 'ChromaAlertActionsRow';
export var useStyles = makeStyles(function () { return ({
    root: {
        marginTop: '0.5rem',
        marginLeft: '-0.625rem',
        marginRight: '-0.625rem',
        '& > button, a': {
            '&:not(:last-child)': {
                marginRight: '0.25rem',
            },
        },
    },
}); }, { name: AlertActionsRowStylesKey });
export var AlertActionsRow = function (_a) {
    var className = _a.className, children = _a.children, rootProps = __rest(_a, ["className", "children"]);
    var classes = useStyles({});
    return (React.createElement("div", __assign({ className: clsx(classes.root, className) }, rootProps), children));
};
