// @lifeomic/chroma-react/esm/components/Alert/Alert.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import * as React from 'react';
import { makeStyles } from '../../styles';
import clsx from 'clsx';
export var AlertStylesKey = 'ChromaAlert';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        alignItems: 'flex-start',
        borderRadius: '0.375rem',
        display: 'flex',
        justifyContent: 'center',
        padding: theme.spacing(2, 3),
    },
    fullWidth: {
        width: '100%',
    },
    info: {
        backgroundColor: theme.palette.blue[50],
        color: theme.palette.primary[800],
    },
    warning: {
        backgroundColor: theme.palette.yellow[100],
        color: theme.palette.yellow[900],
    },
    error: {
        backgroundColor: theme.palette.red[50],
        color: theme.palette.red[900],
    },
    success: {
        backgroundColor: theme.palette.green[100],
        color: theme.palette.green[900],
    },
}); }, { name: AlertStylesKey });
export var Alert = function (_a) {
    var className = _a.className, children = _a.children, fullWidth = _a.fullWidth, _b = _a.statusType, statusType = _b === void 0 ? 'info' : _b, rootProps = __rest(_a, ["className", "children", "fullWidth", "statusType"]);
    var classes = useStyles({});
    return (React.createElement("div", __assign({ className: clsx(classes.root, classes[statusType], fullWidth && classes.fullWidth, className), role: "alert" }, rootProps), children));
};
