// @lifeomic/chroma-react/components/NumberFormat/PhoneNumberFormatField.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PhoneNumberFormatField = exports.PhoneNumberFormatFieldStylesKey = void 0;
require("react-phone-number-input/style.css");
var React = __importStar(require("react"));
var TextField_1 = require("../TextField");
// @ts-ignore The flags module is not typed
var flags_1 = __importDefault(require("react-phone-number-input/flags"));
var react_phone_number_input_1 = __importDefault(require("react-phone-number-input"));
var styles_1 = require("../../styles");
var clsx_1 = __importDefault(require("clsx"));
exports.PhoneNumberFormatFieldStylesKey = 'ChromaPhoneNumberFormatField';
var useStyles = function (_a) {
    var about = _a.about, errorMessage = _a.errorMessage;
    return (0, styles_1.makeStyles)(function (theme) { return ({
        textField: {
            marginTop: theme.spacing(2),
            marginBottom: theme.spacing(1),
        },
        phoneInputRoot: {
            display: 'flex',
            width: '100%',
            position: 'relative',
            '& > .PhoneInputCountry': {
                position: 'absolute',
                bottom: theme.spacing(about || errorMessage ? 5.25 : 2.5),
                left: theme.spacing(1),
                '& .PhoneInputCountryIcon--border': {
                    boxShadow: 'unset',
                    backgroundColor: 'unset',
                },
            },
        },
        phoneInput: {
            '& > input': {
                paddingLeft: theme.spacing(6),
            },
        },
    }); }, { name: exports.PhoneNumberFormatFieldStylesKey })({});
};
var PhoneInputCompatibleChromaInput = React.forwardRef(function (props, ref) {
    var classes = useStyles({});
    var className = props.className, textFieldProps = __rest(props, ["className"]);
    return (React.createElement(TextField_1.TextField, __assign({ fullWidth: true, label: "Phone Number", helpMessage: props.about, ref: ref }, textFieldProps, { className: (0, clsx_1.default)(classes.phoneInput, classes.textField, className) })));
});
var PhoneNumberFormatField = function (_a) {
    var className = _a.className, props = __rest(_a, ["className"]);
    var value = props.value, onChange = props.onChange, errorMessage = props.errorMessage;
    var classes = useStyles(props);
    return (React.createElement(react_phone_number_input_1.default, __assign({ className: (0, clsx_1.default)(classes.phoneInputRoot, className), defaultCountry: "US", flags: flags_1.default }, props, { 
        // @ts-ignore defaultValue has type mismatch
        inputComponent: PhoneInputCompatibleChromaInput, value: value, onChange: onChange, error: errorMessage })));
};
exports.PhoneNumberFormatField = PhoneNumberFormatField;
