// @lifeomic/chroma-react/esm/hooks/events/useInfiniteScroll.js
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
import { useEffect, useRef, useState } from 'react';
import useInterval from '../utils/useInterval';
import useWindowSize from './useWindowSize';
var WINDOW = 'window';
var PARENT = 'parent';
var getParentSize = function (ref) {
    return ref.current.parentNode.getBoundingClientRect();
};
var getMySize = function (ref) {
    return ref.current.getBoundingClientRect();
};
export var helpers = {
    getParentSize: getParentSize,
    getMySize: getMySize,
};
function useInfiniteScroll(_a) {
    var loading = _a.loading, hasNextPage = _a.hasNextPage, loadMore = _a.loadMore, _b = _a.scrollOffset, scrollOffset = _b === void 0 ? 250 : _b, _c = _a.debounceInterval, debounceInterval = _c === void 0 ? 250 : _c, _d = _a.scrollContainer, scrollContainer = _d === void 0 ? WINDOW : _d;
    var ref = useRef();
    var _e = useWindowSize(), windowHeight = _e.height, windowWidth = _e.width;
    var _f = useState(true), listen = _f[0], setListen = _f[1];
    useEffect(function () {
        if (!loading) {
            setListen(true);
        }
    }, [loading]);
    function getBottomOffset() {
        var bottom = helpers.getMySize(ref).bottom;
        var bottomOffset = windowHeight ? bottom - windowHeight : 0;
        if (scrollContainer === PARENT) {
            var parentBottom = helpers.getParentSize(ref).bottom;
            // Distance between bottom of list and its parent
            bottomOffset = bottom - parentBottom;
        }
        return bottomOffset;
    }
    function isParentInView() {
        var parent = ref.current ? ref.current.parentNode : null;
        if (parent) {
            var _a = helpers.getParentSize(ref), left = _a.left, right = _a.right, top_1 = _a.top, bottom = _a.bottom;
            if (windowWidth && left > windowWidth) {
                return false;
            }
            else if (right < 0) {
                return false;
            }
            else if (windowHeight && top_1 > windowHeight) {
                return false;
            }
            else if (bottom < 0) {
                return false;
            }
        }
        return true;
    }
    function listenBottomOffset() {
        return __awaiter(this, void 0, void 0, function () {
            var bottomOffset, validOffset;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (!(listen && !loading && hasNextPage)) return [3 /*break*/, 2];
                        if (!ref.current) return [3 /*break*/, 2];
                        if (scrollContainer === PARENT) {
                            if (!isParentInView()) {
                                // Do nothing if the parent is out of screen
                                return [2 /*return*/];
                            }
                        }
                        bottomOffset = getBottomOffset();
                        validOffset = bottomOffset < scrollOffset;
                        if (!validOffset) return [3 /*break*/, 2];
                        setListen(false);
                        return [4 /*yield*/, loadMore()];
                    case 1:
                        _a.sent();
                        _a.label = 2;
                    case 2: return [2 /*return*/];
                }
            });
        });
    }
    useInterval(function () { return listenBottomOffset(); }, 
    // Stop interval when there is no next page.
    hasNextPage ? debounceInterval : 0);
    return ref;
}
export default useInfiniteScroll;
