// @itwin/itwinui-react/cjs/core/Table/Table.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Table = exports.tableResizeStartAction = void 0;
/*---------------------------------------------------------------------------------------------
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
var react_1 = __importDefault(require("react"));
var classnames_1 = __importDefault(require("classnames"));
var react_table_1 = require("react-table");
var ProgressIndicators_1 = require("../ProgressIndicators");
var utils_1 = require("../utils");
require("@itwin/itwinui-css/css/table.css");
var SortDown_1 = __importDefault(require("@itwin/itwinui-icons-react/cjs/icons/SortDown"));
var SortUp_1 = __importDefault(require("@itwin/itwinui-icons-react/cjs/icons/SortUp"));
var utils_2 = require("./utils");
var TableRowMemoized_1 = require("./TableRowMemoized");
var filters_1 = require("./filters");
var customFilterFunctions_1 = require("./filters/customFilterFunctions");
var hooks_1 = require("./hooks");
var actionHandlers_1 = require("./actionHandlers");
var VirtualScroll_1 = __importDefault(require("../utils/components/VirtualScroll"));
var columns_1 = require("./columns");
var singleRowSelectedAction = 'singleRowSelected';
exports.tableResizeStartAction = 'tableResizeStart';
var tableResizeEndAction = 'tableResizeEnd';
var flattenColumns = function (columns) {
    var flatColumns = [];
    columns.forEach(function (column) {
        flatColumns.push(column);
        if (column.columns) {
            flatColumns.push.apply(flatColumns, flattenColumns(column.columns));
        }
    });
    return flatColumns;
};
/**
 * Table based on [react-table](https://react-table.tanstack.com/docs/api/overview).
 * @example
 * const columns = React.useMemo(() => [
 *  {
 *    Header: 'Header name',
 *    columns: [
 *      {
 *        id: 'name',
 *        Header: 'Name',
 *        accessor: 'name',
 *        width: 90,
 *      },
 *      {
 *        id: 'description',
 *        Header: 'description',
 *        accessor: 'description',
 *        maxWidth: 200,
 *      },
 *      {
 *        id: 'view',
 *        Header: 'view',
 *        Cell: () => {
 *          return <span onClick={onViewClick}>View</span>
 *        },
 *      },
 *    ],
 *  },
 * ], [onViewClick])
 * const data = [
 *  { name: 'Name1', description: 'Description1' },
 *  { name: 'Name2', description: 'Description2' },
 *  { name: 'Name3', description: 'Description3' },
 * ]
 * <Table
 *   columns={columns}
 *   data={data}
 *   emptyTableContent='No data.'
 *   isLoading={false}
 *   isSortable={true}
 * />
 */
var Table = function (props) {
    var _a;
    var data = props.data, columns = props.columns, _b = props.isLoading, isLoading = _b === void 0 ? false : _b, emptyTableContent = props.emptyTableContent, className = props.className, style = props.style, id = props.id, _c = props.isSelectable, isSelectable = _c === void 0 ? false : _c, onSelect = props.onSelect, onRowClick = props.onRowClick, _d = props.selectionMode, selectionMode = _d === void 0 ? 'multi' : _d, _e = props.isSortable, isSortable = _e === void 0 ? false : _e, onSort = props.onSort, stateReducer = props.stateReducer, onBottomReached = props.onBottomReached, onRowInViewport = props.onRowInViewport, _f = props.intersectionMargin, intersectionMargin = _f === void 0 ? 300 : _f, subComponent = props.subComponent, onExpand = props.onExpand, onFilter = props.onFilter, emptyFilteredTableContent = props.emptyFilteredTableContent, filterFunctions = props.filterTypes, expanderCell = props.expanderCell, isRowDisabled = props.isRowDisabled, rowProps = props.rowProps, _g = props.density, density = _g === void 0 ? 'default' : _g, _h = props.selectSubRows, selectSubRows = _h === void 0 ? true : _h, getSubRows = props.getSubRows, _j = props.selectRowOnClick, selectRowOnClick = _j === void 0 ? true : _j, paginatorRenderer = props.paginatorRenderer, _k = props.pageSize, pageSize = _k === void 0 ? 25 : _k, _l = props.isResizable, isResizable = _l === void 0 ? false : _l, _m = props.styleType, styleType = _m === void 0 ? 'default' : _m, _o = props.enableVirtualization, enableVirtualization = _o === void 0 ? false : _o, _p = props.enableColumnReordering, enableColumnReordering = _p === void 0 ? false : _p, rest = __rest(props, ["data", "columns", "isLoading", "emptyTableContent", "className", "style", "id", "isSelectable", "onSelect", "onRowClick", "selectionMode", "isSortable", "onSort", "stateReducer", "onBottomReached", "onRowInViewport", "intersectionMargin", "subComponent", "onExpand", "onFilter", "emptyFilteredTableContent", "filterTypes", "expanderCell", "isRowDisabled", "rowProps", "density", "selectSubRows", "getSubRows", "selectRowOnClick", "paginatorRenderer", "pageSize", "isResizable", "styleType", "enableVirtualization", "enableColumnReordering"]);
    (0, utils_1.useTheme)();
    var _q = react_1.default.useState(), ownerDocument = _q[0], setOwnerDocument = _q[1];
    var defaultColumn = react_1.default.useMemo(function () { return ({
        maxWidth: 0,
        minWidth: 0,
        width: 0,
    }); }, []);
    // useRef prevents from rerendering when one of these callbacks changes
    var onBottomReachedRef = react_1.default.useRef(onBottomReached);
    var onRowInViewportRef = react_1.default.useRef(onRowInViewport);
    react_1.default.useEffect(function () {
        onBottomReachedRef.current = onBottomReached;
        onRowInViewportRef.current = onRowInViewport;
    }, [onBottomReached, onRowInViewport]);
    var hasManualSelectionColumn = react_1.default.useMemo(function () {
        var flatColumns = flattenColumns(columns);
        return flatColumns.some(function (column) { return column.id === columns_1.SELECTION_CELL_ID; });
    }, [columns]);
    var tableStateReducer = react_1.default.useCallback(function (newState, action, previousState, instance) {
        switch (action.type) {
            case react_table_1.actions.toggleSortBy:
                onSort === null || onSort === void 0 ? void 0 : onSort(newState);
                break;
            case react_table_1.actions.setFilter:
                (0, actionHandlers_1.onFilterHandler)(newState, action, previousState, instance, onFilter);
                break;
            case react_table_1.actions.toggleRowExpanded:
            case react_table_1.actions.toggleAllRowsExpanded:
                (0, actionHandlers_1.onExpandHandler)(newState, instance, onExpand);
                break;
            case singleRowSelectedAction: {
                newState = (0, actionHandlers_1.onSingleSelectHandler)(newState, action, instance, onSelect, 
                // If it has manual selection column, then we can't check whether row is disabled
                hasManualSelectionColumn ? undefined : isRowDisabled);
                break;
            }
            case react_table_1.actions.toggleRowSelected:
            case react_table_1.actions.toggleAllRowsSelected:
            case react_table_1.actions.toggleAllPageRowsSelected: {
                (0, actionHandlers_1.onSelectHandler)(newState, instance, onSelect, 
                // If it has manual selection column, then we can't check whether row is disabled
                hasManualSelectionColumn ? undefined : isRowDisabled);
                break;
            }
            case exports.tableResizeStartAction: {
                newState = (0, actionHandlers_1.onTableResizeStart)(newState);
                break;
            }
            case tableResizeEndAction: {
                newState = (0, actionHandlers_1.onTableResizeEnd)(newState, action);
                break;
            }
            default:
                break;
        }
        return stateReducer
            ? stateReducer(newState, action, previousState, instance)
            : newState;
    }, [
        hasManualSelectionColumn,
        isRowDisabled,
        onExpand,
        onFilter,
        onSelect,
        onSort,
        stateReducer,
    ]);
    var filterTypes = react_1.default.useMemo(function () { return (__assign(__assign({}, customFilterFunctions_1.customFilterFunctions), filterFunctions)); }, [filterFunctions]);
    var hasAnySubRows = react_1.default.useMemo(function () {
        return data.some(function (item, index) {
            return getSubRows ? getSubRows(item, index) : item.subRows;
        });
    }, [data, getSubRows]);
    var instance = (0, react_table_1.useTable)(__assign(__assign({ manualPagination: !paginatorRenderer, paginateExpandedRows: false }, props), { columns: columns, defaultColumn: defaultColumn, disableSortBy: !isSortable, stateReducer: tableStateReducer, filterTypes: filterTypes, selectSubRows: selectSubRows, data: data, getSubRows: getSubRows, initialState: __assign({ pageSize: pageSize }, props.initialState) }), react_table_1.useFlexLayout, (0, hooks_1.useResizeColumns)(ownerDocument), react_table_1.useFilters, (0, hooks_1.useSubRowFiltering)(hasAnySubRows), react_table_1.useSortBy, react_table_1.useExpanded, react_table_1.usePagination, react_table_1.useRowSelect, hooks_1.useSubRowSelection, (0, hooks_1.useExpanderCell)(subComponent, expanderCell, isRowDisabled), (0, hooks_1.useSelectionCell)(isSelectable, selectionMode, isRowDisabled), react_table_1.useColumnOrder, (0, hooks_1.useColumnDragAndDrop)(enableColumnReordering));
    var getTableProps = instance.getTableProps, rows = instance.rows, headerGroups = instance.headerGroups, getTableBodyProps = instance.getTableBodyProps, prepareRow = instance.prepareRow, state = instance.state, allColumns = instance.allColumns, filteredFlatRows = instance.filteredFlatRows, dispatch = instance.dispatch, page = instance.page, gotoPage = instance.gotoPage, setPageSize = instance.setPageSize, flatHeaders = instance.flatHeaders;
    var ariaDataAttributes = Object.entries(rest).reduce(function (result, _a) {
        var key = _a[0], value = _a[1];
        if (key.startsWith('data-') || key.startsWith('aria-')) {
            result[key] = value;
        }
        return result;
    }, {});
    var areFiltersSet = allColumns.some(function (column) { return column.filterValue != null && column.filterValue !== ''; });
    var onRowClickHandler = react_1.default.useCallback(function (event, row) {
        var isDisabled = isRowDisabled === null || isRowDisabled === void 0 ? void 0 : isRowDisabled(row.original);
        if (!isDisabled) {
            onRowClick === null || onRowClick === void 0 ? void 0 : onRowClick(event, row);
        }
        if (isSelectable &&
            !isDisabled &&
            selectRowOnClick &&
            !event.isDefaultPrevented()) {
            if (!row.isSelected && (selectionMode === 'single' || !event.ctrlKey)) {
                dispatch({
                    type: singleRowSelectedAction,
                    id: row.id,
                });
            }
            else {
                row.toggleRowSelected(!row.isSelected);
            }
        }
    }, [
        isRowDisabled,
        isSelectable,
        selectRowOnClick,
        selectionMode,
        dispatch,
        onRowClick,
    ]);
    react_1.default.useEffect(function () {
        setPageSize(pageSize);
    }, [pageSize, setPageSize]);
    var paginatorRendererProps = react_1.default.useMemo(function () { return ({
        currentPage: state.pageIndex,
        pageSize: state.pageSize,
        totalRowsCount: rows.length,
        size: density !== 'default' ? 'small' : 'default',
        isLoading: isLoading,
        onPageChange: gotoPage,
        onPageSizeChange: setPageSize,
    }); }, [
        density,
        gotoPage,
        isLoading,
        rows.length,
        setPageSize,
        state.pageIndex,
        state.pageSize,
    ]);
    var columnRefs = react_1.default.useRef({});
    var previousTableWidth = react_1.default.useRef(0);
    var onTableResize = react_1.default.useCallback(function (_a) {
        var width = _a.width;
        if (width === previousTableWidth.current) {
            return;
        }
        previousTableWidth.current = width;
        // Update column widths when table was resized
        flatHeaders.forEach(function (header) {
            if (columnRefs.current[header.id]) {
                header.resizeWidth = columnRefs.current[header.id].getBoundingClientRect().width;
            }
        });
        // If no column was resized then leave table resize handling to the flexbox
        if (Object.keys(state.columnResizing.columnWidths).length === 0) {
            return;
        }
        dispatch({ type: exports.tableResizeStartAction });
    }, [dispatch, state.columnResizing.columnWidths, flatHeaders]);
    var resizeRef = (0, utils_1.useResizeObserver)(onTableResize)[0];
    // Flexbox handles columns resize so we take new column widths before browser repaints.
    react_1.default.useLayoutEffect(function () {
        if (state.isTableResizing) {
            var newColumnWidths_1 = {};
            flatHeaders.forEach(function (column) {
                if (columnRefs.current[column.id]) {
                    newColumnWidths_1[column.id] = columnRefs.current[column.id].getBoundingClientRect().width;
                }
            });
            dispatch({ type: tableResizeEndAction, columnWidths: newColumnWidths_1 });
        }
    });
    var headerRef = react_1.default.useRef(null);
    var bodyRef = react_1.default.useRef(null);
    var getPreparedRow = react_1.default.useCallback(function (index) {
        var row = page[index];
        prepareRow(row);
        return (react_1.default.createElement(TableRowMemoized_1.TableRowMemoized, { row: row, rowProps: rowProps, isLast: index === page.length - 1, onRowInViewport: onRowInViewportRef, onBottomReached: onBottomReachedRef, intersectionMargin: intersectionMargin, state: state, key: row.getRowProps().key, onClick: onRowClickHandler, subComponent: subComponent, isDisabled: !!(isRowDisabled === null || isRowDisabled === void 0 ? void 0 : isRowDisabled(row.original)), tableHasSubRows: hasAnySubRows, tableInstance: instance, expanderCell: expanderCell }));
    }, [
        page,
        expanderCell,
        hasAnySubRows,
        instance,
        intersectionMargin,
        isRowDisabled,
        onRowClickHandler,
        prepareRow,
        rowProps,
        state,
        subComponent,
    ]);
    var virtualizedItemRenderer = react_1.default.useCallback(function (index) { return getPreparedRow(index); }, [getPreparedRow]);
    return (react_1.default.createElement(react_1.default.Fragment, null,
        react_1.default.createElement("div", __assign({ ref: function (element) {
                setOwnerDocument(element === null || element === void 0 ? void 0 : element.ownerDocument);
                if (isResizable) {
                    resizeRef(element);
                }
            }, id: id }, getTableProps({
            className: (0, classnames_1.default)('iui-table', (_a = {}, _a["iui-".concat(density)] = density !== 'default', _a), className),
            style: style,
        }), ariaDataAttributes),
            react_1.default.createElement("div", { className: 'iui-table-header', ref: headerRef }, headerGroups.slice(1).map(function (headerGroup) {
                var headerGroupProps = headerGroup.getHeaderGroupProps({
                    className: 'iui-row',
                });
                return (react_1.default.createElement("div", __assign({}, headerGroupProps, { key: headerGroupProps.key }), headerGroup.headers.map(function (column, index) {
                    var columnProps = column.getHeaderProps(__assign(__assign({}, column.getSortByToggleProps()), { className: (0, classnames_1.default)('iui-cell', { 'iui-actionable': column.canSort }, { 'iui-sorted': column.isSorted }, column.columnClassName), style: __assign(__assign({}, (0, utils_2.getCellStyle)(column, !!state.isTableResizing)), { flexWrap: 'unset' }) }));
                    return (react_1.default.createElement("div", __assign({}, columnProps, column.getDragAndDropProps(), { key: columnProps.key, title: undefined, ref: function (el) {
                            if (el && isResizable) {
                                columnRefs.current[column.id] = el;
                                column.resizeWidth = el.getBoundingClientRect().width;
                            }
                        } }),
                        column.render('Header'),
                        (data.length !== 0 || areFiltersSet) && (react_1.default.createElement(filters_1.FilterToggle, { column: column, ownerDocument: ownerDocument })),
                        data.length !== 0 && column.canSort && (react_1.default.createElement("div", { className: 'iui-cell-end-icon' }, column.isSorted && column.isSortedDesc ? (react_1.default.createElement(SortDown_1.default, { className: 'iui-icon iui-sort', "aria-hidden": true })) : (react_1.default.createElement(SortUp_1.default, { className: 'iui-icon iui-sort', "aria-hidden": true })))),
                        isResizable &&
                            column.isResizerVisible &&
                            index !== headerGroup.headers.length - 1 && (react_1.default.createElement("div", __assign({}, column.getResizerProps(), { className: 'iui-resizer' }),
                            react_1.default.createElement("div", { className: 'iui-resizer-bar' }))),
                        enableColumnReordering && !column.disableReordering && (react_1.default.createElement("div", { className: 'iui-reorder-bar' }))));
                })));
            })),
            react_1.default.createElement("div", __assign({}, getTableBodyProps({
                className: (0, classnames_1.default)('iui-table-body', {
                    'iui-zebra-striping': styleType === 'zebra-rows',
                }),
                style: { outline: 0 },
            }), { ref: bodyRef, onScroll: function () {
                    if (headerRef.current && bodyRef.current) {
                        headerRef.current.scrollLeft = bodyRef.current.scrollLeft;
                    }
                }, tabIndex: -1 }),
                data.length !== 0 && (react_1.default.createElement(react_1.default.Fragment, null, enableVirtualization ? (react_1.default.createElement(VirtualScroll_1.default, { itemsLength: page.length, itemRenderer: virtualizedItemRenderer })) : (page.map(function (_, index) { return getPreparedRow(index); })))),
                isLoading && data.length === 0 && (react_1.default.createElement("div", { className: 'iui-table-empty' },
                    react_1.default.createElement(ProgressIndicators_1.ProgressRadial, { indeterminate: true }))),
                isLoading && data.length !== 0 && (react_1.default.createElement("div", { className: 'iui-row' },
                    react_1.default.createElement("div", { className: 'iui-cell', style: { justifyContent: 'center' } },
                        react_1.default.createElement(ProgressIndicators_1.ProgressRadial, { indeterminate: true, size: 'small', style: { float: 'none', marginLeft: 0 } })))),
                !isLoading && data.length === 0 && !areFiltersSet && (react_1.default.createElement("div", { className: 'iui-table-empty' },
                    react_1.default.createElement("div", null, emptyTableContent))),
                !isLoading &&
                    (data.length === 0 || filteredFlatRows.length === 0) &&
                    areFiltersSet && (react_1.default.createElement("div", { className: 'iui-table-empty' },
                    react_1.default.createElement("div", null, emptyFilteredTableContent)))), paginatorRenderer === null || paginatorRenderer === void 0 ? void 0 :
            paginatorRenderer(paginatorRendererProps))));
};
exports.Table = Table;
exports.default = exports.Table;
