// @lifeomic/chroma-react/esm/components/Avatar/AvatarBadge.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { makeStyles } from '../../styles';
import { useAvatarSize } from './AvatarSizeContext';
export var AvatarBadgeStylesKey = 'ChromaAvatarBadge';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        alignItems: 'center',
        backgroundColor: theme.palette.red.main,
        border: "".concat(theme.pxToRem(1), " solid ").concat(theme.palette.black[50]),
        borderRadius: 9999,
        bottom: 0,
        color: theme.palette.common.white,
        display: 'flex',
        fontWeight: theme.typography.fontWeightBold,
        justifyContent: 'center',
        position: 'absolute',
        right: 0,
        transform: 'translate(25%, 25%)',
        userSelect: 'none',
    },
    size0: {
        height: theme.pxToRem(12),
        width: theme.pxToRem(12),
    },
    size1: {
        fontSize: theme.pxToRem(8),
        height: theme.pxToRem(16),
        width: theme.pxToRem(16),
    },
    size2: {
        fontSize: theme.pxToRem(10),
        height: theme.pxToRem(20),
        width: theme.pxToRem(20),
    },
}); }, { name: AvatarBadgeStylesKey });
export var AvatarBadge = function (_a) {
    var _b;
    var className = _a.className, count = _a.count, isVisible = _a.isVisible, rootProps = __rest(_a, ["className", "count", "isVisible"]);
    var classes = useStyles({});
    var size = useAvatarSize().size;
    if (!count && !isVisible) {
        return null;
    }
    return (React.createElement("span", __assign({ "aria-label": count ? "".concat(count, " notifications available") : 'notifications available', role: "status", className: clsx(classes.root, (_b = {},
            _b[classes.size0] = size === 0,
            _b[classes.size1] = size === 1,
            _b[classes.size2] = size === 2,
            _b), className) }, rootProps), size !== 0 ? count : null));
};
