// @lifeomic/chroma-react/components/PageLayout/PageLayout.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PageLayout = exports.useStyles = exports.PageLayoutStylesKey = void 0;
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
var PageLayoutDefaultHero_svg_1 = __importDefault(require("../../assets/PageLayoutDefaultHero.svg"));
var styles_1 = require("../../styles");
var PageHeader_1 = require("../PageHeader");
exports.PageLayoutStylesKey = 'ChromaPageLayout';
exports.useStyles = (0, styles_1.makeStyles)(function (theme) { return ({
    root: {
        display: 'flex',
        flexDirection: 'column',
        flex: 1,
        minHeight: '100%',
        marginRight: 'auto',
        marginLeft: 'auto',
        paddingLeft: theme.spacing(4),
        paddingRight: theme.spacing(4),
    },
    maxWidth: {
        maxWidth: theme.pxToRem(1344),
    },
    backgroundCover: {
        position: 'relative',
        // Negative margin so that it can stretch
        // outside of the default padding specified by `root`.
        marginLeft: -theme.spacing(4),
        marginRight: -theme.spacing(4),
        background: "url(".concat(PageLayoutDefaultHero_svg_1.default, ") ").concat(theme.palette.primary.main),
        backgroundSize: 'cover',
        marginBottom: theme.spacing(2),
    },
    headerBackgroundPadding: {
        paddingTop: theme.spacing(3),
        paddingLeft: theme.spacing(4),
        paddingRight: theme.spacing(4),
        paddingBottom: theme.spacing(3),
    },
    disclaimer: {
        paddingBottom: theme.spacing(2),
    },
    tabs: {
        padding: theme.spacing(0),
    },
    sidebar: {
        width: theme.pxToRem(192),
        marginRight: theme.spacing(3),
        flexGrow: 0,
        flexShrink: 0,
    },
    layoutContainer: {
        display: 'flex',
        height: '100%',
        marginBottom: theme.spacing(4),
    },
    content: {
        flex: 1,
        width: '100%',
        // Fix for firefox so the content doesn't overflow
        maxWidth: '100%',
    },
}); }, { name: exports.PageLayoutStylesKey });
var ConditionalWrapper = function (_a) {
    var condition = _a.condition, wrapper = _a.wrapper, children = _a.children;
    return (condition ? wrapper(children) : children);
};
exports.PageLayout = React.forwardRef(function (_a, ref) {
    var backgroundImage = _a.backgroundImage, _b = _a.headerAlign, headerAlign = _b === void 0 ? 'center' : _b, headerCenter = _a.headerCenter, headerCenterClassName = _a.headerCenterClassName, headerActionsClassName = _a.headerActionsClassName, children = _a.children, additionalClasses = _a.classes, className = _a.className, disclaimer = _a.disclaimer, headerActions = _a.headerActions, sidebar = _a.sidebar, title = _a.title, tabs = _a.tabs, _c = _a.maxWidth, maxWidth = _c === void 0 ? true : _c, _d = _a.applyBackgroundCover, applyBackgroundCover = _d === void 0 ? false : _d, left = _a.left, leftClassName = _a.leftClassName, rootProps = __rest(_a, ["backgroundImage", "headerAlign", "headerCenter", "headerCenterClassName", "headerActionsClassName", "children", "classes", "className", "disclaimer", "headerActions", "sidebar", "title", "tabs", "maxWidth", "applyBackgroundCover", "left", "leftClassName"]);
    var classes = (0, exports.useStyles)({ classes: additionalClasses });
    return (React.createElement("div", __assign({ ref: ref, className: (0, clsx_1.default)(classes.root, maxWidth && !applyBackgroundCover && classes.maxWidth, className) }, rootProps),
        React.createElement(ConditionalWrapper, { condition: applyBackgroundCover, wrapper: function (children) { return (React.createElement("div", { className: classes.backgroundCover, style: backgroundImage
                    ? {
                        backgroundImage: "url(".concat(backgroundImage, ")"),
                    }
                    : undefined }, children)); } },
            React.createElement(PageHeader_1.PageHeader, { align: headerAlign, className: applyBackgroundCover ? classes.headerBackgroundPadding : '', title: title, center: headerCenter, centerClassName: headerCenterClassName, margin: "none", actions: headerActions, actionsClassName: headerActionsClassName, color: applyBackgroundCover ? 'inverse' : 'default', left: left, leftClassName: leftClassName })),
        !!disclaimer && React.createElement("div", { className: classes.disclaimer }, disclaimer),
        !!tabs && React.createElement("div", { className: classes.tabs }, tabs),
        React.createElement("div", { className: classes.layoutContainer },
            !!sidebar && React.createElement("div", { className: classes.sidebar }, sidebar),
            React.createElement("div", { className: classes.content }, children))));
});
