// @lifeomic/chroma-react/components/SmallTile/SmallTileFooter.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
var styles_1 = require("../../styles");
var Text_1 = require("../Text");
exports.SmallTileFooterStylesKey = 'ChromaSmallTileFooter';
exports.useStyles = styles_1.makeStyles(function (theme) { return ({
    root: {
        display: 'flex',
        padding: theme.spacing(1.375, 1.375, 1.375, 1.5),
        borderTop: "1px solid " + theme.palette.divider,
    },
    text: {
        color: theme.palette.black[500],
        letterSpacing: 'unset',
        lineHeight: 1.5,
    },
    justifyStart: {
        justifyContent: 'flex-start',
    },
    justifyEnd: {
        justifyContent: 'flex-end',
    },
    justifyCenter: {
        justifyContent: 'center',
    },
    justifyBetween: {
        justifyContent: 'space-between',
    },
}); }, { name: exports.SmallTileFooterStylesKey });
exports.SmallTileFooter = React.forwardRef(function (_a, ref) {
    var _b;
    var children = _a.children, className = _a.className, _c = _a.justify, justify = _c === void 0 ? 'space-between' : _c, text = _a.text, rootProps = __rest(_a, ["children", "className", "justify", "text"]);
    var classes = exports.useStyles({});
    return (React.createElement("div", __assign({ ref: ref, className: clsx_1.default(classes.root, (_b = {},
            _b[classes.justifyBetween] = justify === 'space-between',
            _b[classes.justifyCenter] = justify === 'center',
            _b[classes.justifyEnd] = justify === 'flex-end',
            _b[classes.justifyStart] = justify === 'flex-start',
            _b), className) }, rootProps),
        !!text && (React.createElement(Text_1.Text, { size: "caption", className: classes.text }, text)),
        children));
});
