// @lifeomic/chroma-react/components/SmallTile/SmallTile.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
var styles_1 = require("../../styles");
exports.SmallTileStylesKey = 'ChromaSmallTile';
exports.useStyles = styles_1.makeStyles(function (theme) { return ({
    root: {
        width: theme.pxToRem(172),
        height: theme.pxToRem(115),
        borderRadius: 5,
        background: theme.palette.common.white,
        boxShadow: theme.boxShadows.table,
    },
    cursorPointer: {
        cursor: 'pointer',
        transition: 'transform 0.5s ease, box-shadow 0.25s ease-out',
        '&:hover, &:focus': {
            outline: 'none',
            transform: 'translate3d(0, -2px, 0)',
            boxShadow: theme.boxShadows.elevatedContent,
            '& > * > p': {
                transition: 'color 0.5s ease',
                color: theme.palette.primary.main,
            },
        },
    },
}); }, { name: exports.SmallTileStylesKey });
exports.SmallTile = React.forwardRef(function (_a, ref) {
    var children = _a.children, className = _a.className, onClick = _a.onClick, rootProps = __rest(_a, ["children", "className", "onClick"]);
    var classes = exports.useStyles({});
    return (
    // We conditionally add the role if `onClick` is provided
    // eslint-disable-next-line jsx-a11y/no-static-element-interactions
    React.createElement("div", __assign({ ref: ref, className: clsx_1.default(classes.root, className, !!onClick && classes.cursorPointer), role: onClick ? 'button' : undefined, tabIndex: onClick ? 0 : undefined, onClick: onClick }, rootProps), children));
});
