// @lifeomic/chroma-react/components/SmallTile/SmallTileContent.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
var styles_1 = require("../../styles");
var Text_1 = require("../Text");
exports.SmallTileContentStylesKey = 'ChromaSmallTileContent';
exports.useStyles = styles_1.makeStyles(function (theme) { return ({
    root: {
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        textAlign: 'center',
        height: theme.pxToRem(77),
    },
    text: {
        fontWeight: theme.typography.fontWeightBold,
        letterSpacing: '0.4px',
        wordBreak: 'break-word',
        marginLeft: theme.spacing(1),
        marginRight: theme.spacing(1),
        lineHeight: 1.66,
    },
    fullHeight: {
        height: '100%',
    },
}); }, { name: exports.SmallTileContentStylesKey });
exports.SmallTileContent = React.forwardRef(function (_a, ref) {
    var _b = _a.fullHeight, fullHeight = _b === void 0 ? false : _b, className = _a.className, children = _a.children, text = _a.text, rootProps = __rest(_a, ["fullHeight", "className", "children", "text"]);
    var classes = exports.useStyles({});
    return (React.createElement("div", __assign({ ref: ref, className: clsx_1.default(classes.root, fullHeight && classes.fullHeight, className) }, rootProps),
        !!text && (React.createElement(Text_1.Text, { size: "caption", className: classes.text }, text)),
        children));
});
