// @lifeomic/chroma-react/components/KeymapHelp/KeymapHelp.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __spreadArray = (this && this.__spreadArray) || function (to, from, pack) {
    if (pack || arguments.length === 2) for (var i = 0, l = from.length, ar; i < l; i++) {
        if (ar || !(i in from)) {
            if (!ar) ar = Array.prototype.slice.call(from, 0, i);
            ar[i] = from[i];
        }
    }
    return to.concat(ar || Array.prototype.slice.call(from));
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeymapHelp = exports.useStyles = exports.KeymapHelpStylesKey = void 0;
var React = __importStar(require("react"));
var react_hotkeys_1 = require("react-hotkeys");
var styles_1 = require("../../styles");
var Button_1 = require("../Button");
var Chip_1 = require("../Chip");
var Modal_1 = require("../Modal");
var KEY_MAP = {
    SHOW_KEYMAP_HELP: 'shift+?',
};
var BASE_KEY_MAP_DOCS = [
    {
        sequences: ['shift+?'],
        description: 'Show keyboard shortcuts',
    },
    {
        sequences: ['['],
        description: 'Toggle sidebar',
    },
];
exports.KeymapHelpStylesKey = 'ChromaKeymapHelp';
exports.useStyles = (0, styles_1.makeStyles)(function (theme) { return ({
    root: {},
    sequence: {
        paddingLeft: theme.spacing(3.75),
    },
}); }, { name: exports.KeymapHelpStylesKey });
var KeymapHelp = function (_a) {
    var closeButtonText = _a.closeButtonText, keyMapDocs = _a.keyMapDocs, keyListenerTestId = _a.keyListenerTestId;
    var classes = (0, exports.useStyles)({});
    var _b = React.useState(false), isOpen = _b[0], setIsOpen = _b[1];
    var onClose = function () {
        setIsOpen(false);
    };
    var HANDLERS = {
        SHOW_KEYMAP_HELP: function () { return setIsOpen(true); },
    };
    var docs = __spreadArray([], BASE_KEY_MAP_DOCS, true);
    if (keyMapDocs) {
        keyMapDocs.forEach(function (x) { return docs.push(x); });
    }
    return (React.createElement(React.Fragment, null,
        React.createElement(react_hotkeys_1.GlobalHotKeys, { keyMap: KEY_MAP, handlers: HANDLERS },
            React.createElement("span", { "data-testid": keyListenerTestId })),
        React.createElement(Modal_1.Modal, { isOpen: isOpen, title: "Keyboard shortcuts", onDismiss: onClose, actions: React.createElement(Button_1.Button, { variant: "contained", onClick: onClose }, closeButtonText || 'Close') },
            React.createElement(React.Fragment, null,
                React.createElement("table", null,
                    React.createElement("tbody", null, docs.map(function (_a) {
                        var sequences = _a.sequences, description = _a.description;
                        var name = sequences.join('|');
                        return (React.createElement("tr", { key: name },
                            React.createElement("td", null, description),
                            React.createElement("td", { className: classes.sequence }, sequences.map(function (sequence) { return (React.createElement(Chip_1.Chip, { key: sequence, label: sequence })); }))));
                    })))))));
};
exports.KeymapHelp = KeymapHelp;
