// @lifeomic/chroma-react/components/KeymapHelp/KeymapHelp.js
"use strict";
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var react_hotkeys_1 = require("react-hotkeys");
var styles_1 = require("../../styles");
var Button_1 = require("../Button");
var Chip_1 = require("../Chip");
var Modal_1 = require("../Modal");
var KEY_MAP = {
    SHOW_KEYMAP_HELP: 'shift+?',
};
var BASE_KEY_MAP_DOCS = [
    {
        sequences: ['shift+?'],
        description: 'Show keyboard shortcuts',
    },
    {
        sequences: ['['],
        description: 'Toggle sidebar',
    },
];
exports.KeymapHelpStylesKey = 'ChromaKeymapHelp';
exports.useStyles = styles_1.makeStyles(function (theme) { return ({
    root: {},
    sequence: {
        paddingLeft: theme.spacing(3.75),
    },
}); }, { name: exports.KeymapHelpStylesKey });
exports.KeymapHelp = function (_a) {
    var closeButtonText = _a.closeButtonText, keyMapDocs = _a.keyMapDocs;
    var classes = exports.useStyles({});
    var _b = React.useState(false), isOpen = _b[0], setIsOpen = _b[1];
    var onClose = function () {
        setIsOpen(false);
    };
    var HANDLERS = {
        SHOW_KEYMAP_HELP: function () { return setIsOpen(true); },
    };
    var docs = __spreadArrays(BASE_KEY_MAP_DOCS);
    if (keyMapDocs) {
        keyMapDocs.forEach(function (x) { return docs.push(x); });
    }
    return (React.createElement(React.Fragment, null,
        React.createElement(react_hotkeys_1.GlobalHotKeys, { keyMap: KEY_MAP, handlers: HANDLERS },
            React.createElement("span", null)),
        React.createElement(Modal_1.Modal, { isOpen: isOpen, title: "Keyboard shortcuts", onDismiss: onClose, actions: React.createElement(Button_1.Button, { variant: "contained", onClick: onClose }, closeButtonText || 'Close') },
            React.createElement(React.Fragment, null,
                React.createElement("table", null,
                    React.createElement("tbody", null, docs.map(function (_a) {
                        var sequences = _a.sequences, description = _a.description;
                        var name = sequences.join('|');
                        return (React.createElement("tr", { key: name },
                            React.createElement("td", null, description),
                            React.createElement("td", { className: classes.sequence }, sequences.map(function (sequence) { return (React.createElement(Chip_1.Chip, { key: sequence, label: sequence })); }))));
                    })))))));
};
