// @lifeomic/chroma-react/components/Tabs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TabPanelStylesKey = exports.TabPanel = exports.TabListStylesKey = exports.TabList = exports.TabStylesKey = exports.Tab = exports.Tabs = void 0;
var Tabs_1 = require("./Tabs");
Object.defineProperty(exports, "Tabs", { enumerable: true, get: function () { return Tabs_1.Tabs; } });
var Tab_1 = require("./Tab");
Object.defineProperty(exports, "Tab", { enumerable: true, get: function () { return Tab_1.Tab; } });
Object.defineProperty(exports, "TabStylesKey", { enumerable: true, get: function () { return Tab_1.TabStylesKey; } });
var TabList_1 = require("./TabList");
Object.defineProperty(exports, "TabList", { enumerable: true, get: function () { return TabList_1.TabList; } });
Object.defineProperty(exports, "TabListStylesKey", { enumerable: true, get: function () { return TabList_1.TabListStylesKey; } });
var TabPanel_1 = require("./TabPanel");
Object.defineProperty(exports, "TabPanel", { enumerable: true, get: function () { return TabPanel_1.TabPanel; } });
Object.defineProperty(exports, "TabPanelStylesKey", { enumerable: true, get: function () { return TabPanel_1.TabPanelStylesKey; } });
