// @lifeomic/chroma-react/components/Tabs/TabList.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TabList = exports.useStyles = exports.TabListStylesKey = void 0;
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
var Tab_1 = require("reakit/Tab");
var styles_1 = require("../../styles");
var TabsContext_1 = require("./TabsContext");
exports.TabListStylesKey = 'ChromaTabList';
exports.useStyles = (0, styles_1.makeStyles)(function (theme) { return ({
    root: {
        display: 'flex',
        flexDirection: 'row',
        borderBottom: "1px solid ".concat(theme.palette.divider),
        width: '100%',
        overflow: 'hidden',
    },
    pill: {
        background: 'rgba(132, 137, 166, 0.15)',
        borderRadius: theme.pxToRem(20),
        border: 'solid 1px transparent',
        display: 'inline-flex',
        flexDirection: 'row',
        margin: 0,
        maxHeight: theme.pxToRem(35),
        overflow: 'hidden',
        padding: theme.spacing(0.25),
        width: 'auto',
    },
    fullWidth: {
        width: '100%',
    },
}); }, { name: exports.TabListStylesKey });
var TabList = function (_a) {
    var _b;
    var ariaLabel = _a["aria-label"], className = _a.className, rootProps = __rest(_a, ['aria-label', "className"]);
    var classes = (0, exports.useStyles)({});
    var _c = React.useContext(TabsContext_1.TabsContext), variant = _c.variant, fullWidth = _c.fullWidth, tabState = _c.tabState;
    return (React.createElement(Tab_1.TabList, __assign({}, tabState, { "aria-label": ariaLabel, className: (0, clsx_1.default)(classes.root, className, (_b = {},
            _b[classes.pill] = variant === 'pill',
            _b[classes.fullWidth] = fullWidth,
            _b)) }, rootProps)));
};
exports.TabList = TabList;
