// @lifeomic/chroma-react/esm/components/SmallTile/SmallTileFooter.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { makeStyles } from '../../styles';
import { Text } from '../Text';
export var SmallTileFooterStylesKey = 'ChromaSmallTileFooter';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        display: 'flex',
        padding: theme.spacing(1.375, 1.375, 1.375, 1.5),
        borderTop: "1px solid ".concat(theme.palette.divider),
    },
    text: {
        color: theme.palette.black[500],
        letterSpacing: 'unset',
        lineHeight: 1.5,
    },
    justifyStart: {
        justifyContent: 'flex-start',
    },
    justifyEnd: {
        justifyContent: 'flex-end',
    },
    justifyCenter: {
        justifyContent: 'center',
    },
    justifyBetween: {
        justifyContent: 'space-between',
    },
}); }, { name: SmallTileFooterStylesKey });
export var SmallTileFooter = React.forwardRef(function (_a, ref) {
    var _b;
    var children = _a.children, className = _a.className, _c = _a.justify, justify = _c === void 0 ? 'space-between' : _c, text = _a.text, rootProps = __rest(_a, ["children", "className", "justify", "text"]);
    var classes = useStyles({});
    return (React.createElement("div", __assign({ ref: ref, className: clsx(classes.root, (_b = {},
            _b[classes.justifyBetween] = justify === 'space-between',
            _b[classes.justifyCenter] = justify === 'center',
            _b[classes.justifyEnd] = justify === 'flex-end',
            _b[classes.justifyStart] = justify === 'flex-start',
            _b), className) }, rootProps),
        !!text && (React.createElement(Text, { size: "caption", className: classes.text }, text)),
        children));
});
