// @lifeomic/chroma-react/esm/components/SmallTile/SmallTileContent.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { makeStyles } from '../../styles';
import { Text } from '../Text';
export var SmallTileContentStylesKey = 'ChromaSmallTileContent';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        textAlign: 'center',
        height: theme.pxToRem(77),
    },
    text: {
        fontWeight: theme.typography.fontWeightBold,
        letterSpacing: '0.4px',
        wordBreak: 'break-word',
        marginLeft: theme.spacing(1),
        marginRight: theme.spacing(1),
        lineHeight: 1.66,
    },
    fullHeight: {
        height: '100%',
    },
}); }, { name: SmallTileContentStylesKey });
export var SmallTileContent = React.forwardRef(function (_a, ref) {
    var _b = _a.fullHeight, fullHeight = _b === void 0 ? false : _b, className = _a.className, children = _a.children, text = _a.text, rootProps = __rest(_a, ["fullHeight", "className", "children", "text"]);
    var classes = useStyles({});
    return (React.createElement("div", __assign({ ref: ref, className: clsx(classes.root, fullHeight && classes.fullHeight, className) }, rootProps),
        !!text && (React.createElement(Text, { size: "caption", className: classes.text }, text)),
        children));
});
