// @lifeomic/chroma-react/esm/components/SmallTile/SmallTile.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { makeStyles } from '../../styles';
export var SmallTileStylesKey = 'ChromaSmallTile';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        width: theme.pxToRem(172),
        height: theme.pxToRem(115),
        borderRadius: 5,
        background: theme.palette.common.white,
        boxShadow: theme.boxShadows.table,
    },
    cursorPointer: {
        cursor: 'pointer',
        transition: 'transform 0.5s ease, box-shadow 0.25s ease-out',
        '&:hover, &:focus': {
            outline: 'none',
            transform: 'translate3d(0, -2px, 0)',
            boxShadow: theme.boxShadows.elevatedContent,
            '& > * > p': {
                transition: 'color 0.5s ease',
                color: theme.palette.primary.main,
            },
        },
    },
}); }, { name: SmallTileStylesKey });
export var SmallTile = React.forwardRef(function (_a, ref) {
    var children = _a.children, className = _a.className, onClick = _a.onClick, rootProps = __rest(_a, ["children", "className", "onClick"]);
    var classes = useStyles({});
    return (
    // We conditionally add the role if `onClick` is provided
    // eslint-disable-next-line jsx-a11y/no-static-element-interactions
    React.createElement("div", __assign({ ref: ref, className: clsx(classes.root, className, !!onClick && classes.cursorPointer), role: onClick ? 'button' : undefined, tabIndex: onClick ? 0 : undefined, onClick: onClick }, rootProps), children));
});
