// @lifeomic/chroma-react/esm/components/DotLoader/DotLoader.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { makeStyles } from '../../styles';
import { AnimatePresence, motion } from 'framer-motion';
export var DotLoaderStylesKey = 'ChromaDotLoader';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        height: theme.pxToRem(100),
        width: theme.pxToRem(100),
    },
    size0: {
        height: theme.pxToRem(25),
        width: theme.pxToRem(25),
    },
    size1: {
        height: theme.pxToRem(50),
        width: theme.pxToRem(50),
    },
    '@keyframes bounce': {
        '0%': {
            transform: 'translate3d(0, 0, 0)',
        },
        '50%': {
            transform: 'translate3d(0, 1rem, 0)',
        },
        '100%': {
            transform: 'translate3d(0, 0, 0)',
        },
    },
    bounce1: {
        animation: '$bounce 1s 0s infinite',
        fill: theme.palette.red.main,
    },
    bounce2: {
        animation: '$bounce 1s 0.2s infinite',
        fill: '#F7C44D',
    },
    bounce3: {
        animation: '$bounce 1s 0.4s infinite',
        fill: '#09BEEF',
    },
    bounce4: {
        animation: '$bounce 1s 0.6s infinite',
        fill: theme.palette.green.light,
    },
}); }, { name: DotLoaderStylesKey });
export var DotLoader = function (_a) {
    var _b;
    var size = _a.size, dotStyle = _a.dotStyle, rootProps = __rest(_a, ["size", "dotStyle"]);
    var classes = useStyles({});
    return (React.createElement(AnimatePresence, null,
        React.createElement(motion.svg, __assign({ role: "progressbar", version: "1.1", xmlns: "http://www.w3.org/2000/svg", x: "0px", y: "0px", viewBox: "0 0 72 78", className: clsx(classes.root, (_b = {},
                _b[classes.size0] = size === 0,
                _b[classes.size1] = size === 1,
                _b)), initial: { opacity: 0 }, animate: { opacity: 1, transition: { duration: 0.3 } }, exit: { opacity: 0, transition: { duration: 0.25 } } }, rootProps),
            React.createElement("circle", { stroke: "none", cx: "6", cy: "28", r: "6", className: clsx(classes.bounce1, dotStyle) }),
            React.createElement("circle", { stroke: "none", cx: "26", cy: "28", r: "6", className: clsx(classes.bounce2, dotStyle) }),
            React.createElement("circle", { stroke: "none", cx: "46", cy: "28", r: "6", className: clsx(classes.bounce3, dotStyle) }),
            React.createElement("circle", { stroke: "none", cx: "66", cy: "28", r: "6", className: clsx(classes.bounce4, dotStyle) }))));
};
