// @itwin/itwinui-illustrations-react/cjs/illustrations/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SvgRedirect = exports.SvgError = exports.Svg503 = exports.Svg502 = exports.Svg500 = exports.Svg404 = exports.Svg403 = exports.Svg401 = void 0;
var _401_1 = require("./401");
Object.defineProperty(exports, "Svg401", { enumerable: true, get: function () { return __importDefault(_401_1).default; } });
var _403_1 = require("./403");
Object.defineProperty(exports, "Svg403", { enumerable: true, get: function () { return __importDefault(_403_1).default; } });
var _404_1 = require("./404");
Object.defineProperty(exports, "Svg404", { enumerable: true, get: function () { return __importDefault(_404_1).default; } });
var _500_1 = require("./500");
Object.defineProperty(exports, "Svg500", { enumerable: true, get: function () { return __importDefault(_500_1).default; } });
var _502_1 = require("./502");
Object.defineProperty(exports, "Svg502", { enumerable: true, get: function () { return __importDefault(_502_1).default; } });
var _503_1 = require("./503");
Object.defineProperty(exports, "Svg503", { enumerable: true, get: function () { return __importDefault(_503_1).default; } });
var Error_1 = require("./Error");
Object.defineProperty(exports, "SvgError", { enumerable: true, get: function () { return __importDefault(Error_1).default; } });
var Redirect_1 = require("./Redirect");
Object.defineProperty(exports, "SvgRedirect", { enumerable: true, get: function () { return __importDefault(Redirect_1).default; } });
