// @lifeomic/chroma-react/esm/components/Header/Header.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { makeStyles } from '../../styles';
export var headerHeight = '3.125rem';
export var HeaderStylesKey = 'ChromaHeader';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        alignItems: 'center',
        backgroundColor: theme.palette.common.white,
        boxShadow: '0px 4px 54px rgba(0, 0, 0, 0.12);',
        display: 'flex',
        flex: 1,
        height: headerHeight,
        left: 'auto',
        minHeight: headerHeight,
        paddingLeft: '1rem',
        paddingRight: '1rem',
        position: 'relative',
        right: 0,
        top: 0,
        width: '100%',
        zIndex: 1100,
    },
    container: {
        display: 'flex',
        alignItems: 'center',
    },
    logo: {
        flexGrow: 0,
        flexShrink: 0,
        marginRight: theme.spacing(1.5),
    },
    centerLogo: {
        position: 'absolute',
        left: '50%',
        transform: 'translateX(-50%)',
        margin: 0,
    },
    left: {
        flex: 1,
        marginRight: theme.spacing(1.5),
    },
    right: {
        flexGrow: 0,
        flexShrink: 1,
    },
}); }, { name: HeaderStylesKey });
export var Header = React.forwardRef(function (_a, ref) {
    var logo = _a.logo, left = _a.left, right = _a.right, className = _a.className, centerLogo = _a.centerLogo, rootProps = __rest(_a, ["logo", "left", "right", "className", "centerLogo"]);
    var classes = useStyles({});
    return (React.createElement("header", __assign({ ref: ref, className: clsx(classes.root, className), role: "banner" }, rootProps),
        !!logo && (React.createElement("div", { role: "presentation", className: clsx(classes.container, classes.logo, centerLogo && classes.centerLogo) }, logo)),
        React.createElement("div", { className: clsx(classes.container, classes.left) }, left),
        React.createElement("div", { className: clsx(classes.container, classes.right) }, right)));
});
