// @lifeomic/chroma-react/esm/styles/createTypography.js
import deepmerge from 'deepmerge';
export var fontFamilies = {
    sansSerif: '"Lato", Helvetica, Roboto, Arial, "Lucida Grande", sans-serif',
    monospace: '"Ubuntu Mono", Consolas, "Lucida Console", Monaco, Courier, monospace',
};
export var fontWeights = {
    lighter: 100,
    light: 300,
    regular: 400,
    bold: 700,
    bolder: 900,
};
export var fontSizes = {
    h1: '3rem',
    h2: '2.25rem',
    h3: '1.875rem',
    h4: '1.5rem',
    h5: '1.25rem',
    h6: '1.125rem',
    body: '1rem',
    button: '0.75rem',
    subtitle: '0.875rem',
    caption: '0.75rem',
    tooltip: '0.6875rem',
    code: '1rem',
};
var baseTypography = {
    fontFamily: fontFamilies.sansSerif,
    fontFamilyMonospace: fontFamilies.monospace,
    fontWeightLighter: fontWeights.lighter,
    fontWeightLight: fontWeights.light,
    fontWeightRegular: fontWeights.regular,
    fontWeightMedium: fontWeights.regular,
    fontWeightBold: fontWeights.bold,
    fontWeightBolder: fontWeights.bolder,
    h1: {
        fontSize: fontSizes.h1,
    },
    h2: {
        fontSize: fontSizes.h2,
    },
    h3: {
        fontSize: fontSizes.h3,
    },
    h4: {
        fontSize: fontSizes.h4,
    },
    h5: {
        fontSize: fontSizes.h5,
    },
    h6: {
        fontSize: fontSizes.h6,
    },
    body1: {
        fontSize: fontSizes.body,
    },
    body2: {
        fontSize: fontSizes.subtitle,
    },
    subtitle1: {
        fontSize: fontSizes.subtitle,
    },
    subtitle2: {
        fontSize: fontSizes.subtitle,
    },
    button: {
        textTransform: 'initial',
        fontSize: fontSizes.button,
        fontWeight: fontWeights.bold,
        letterSpacing: 0.5,
        lineHeight: 1.4,
    },
    caption: {
        fontSize: fontSizes.caption,
    },
};
export var createTypography = function (typography) {
    if (typography === void 0) { typography = {}; }
    return deepmerge(baseTypography, typography, { clone: false });
};
