// @lifeomic/chroma-react/esm/styles/createPalette.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import deepmerge from 'deepmerge';
import black from '../colors/black';
import blue from '../colors/blue';
import charcoal from '../colors/charcoal';
import darkGraphite from '../colors/darkGraphite';
import graphite from '../colors/graphite';
import green from '../colors/green';
import grey from '../colors/grey';
import navy from '../colors/navy';
import orange from '../colors/orange';
import purple from '../colors/purple';
import red from '../colors/red';
import text from '../colors/text';
import yellow from '../colors/yellow';
var baseMuiPalette = {
    primary: blue,
    secondary: green,
    error: red,
    type: 'light',
    common: {
        black: black.main,
        white: '#ffffff',
    },
    grey: grey,
    text: text,
    divider: grey[400],
    background: {
        paper: grey[50],
        default: '#f5f8fa',
    },
    negative: {
        main: red.main,
        dark: red.dark,
    },
    positive: {
        main: green.main,
        dark: green.dark,
    },
};
var basePalette = __assign({ black: black, blue: blue, charcoal: charcoal, darkGraphite: darkGraphite, graphite: graphite, green: green, navy: navy, orange: orange, purple: purple, red: red, yellow: yellow }, baseMuiPalette);
export var createPalette = function (palette) {
    if (palette === void 0) { palette = {}; }
    return deepmerge(basePalette, palette, { clone: false });
};
