// @lifeomic/chroma-react/components/LayoutManager/LayoutManagerContext.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.useLayoutManager = exports.LayoutManagerContext = exports.LayoutManagerStoreKey = void 0;
var React = __importStar(require("react"));
var canAccessLocalStorage_1 = require("./canAccessLocalStorage");
exports.LayoutManagerStoreKey = '@@lifeomic/store/layoutSidebarCollapsed';
exports.LayoutManagerContext = React.createContext({
    isSidebarCollapseDisabled: false,
    isSidebarCollapsed: !!((0, canAccessLocalStorage_1.canAccessLocalStorage)() &&
        localStorage.getItem(exports.LayoutManagerStoreKey) === 'true'),
    toggleSidebarCollapsed: function () { return undefined; },
});
var useLayoutManager = function () { return React.useContext(exports.LayoutManagerContext); };
exports.useLayoutManager = useLayoutManager;
