// @lifeomic/chroma-react/esm/components/Link/Link.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { NavLink, } from 'react-router-dom';
import { makeStyles } from '../../styles';
export var LinkStylesKey = 'ChromaLink';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        color: 'rgba(0, 83, 154, 0.9)',
        transition: 'color 0.25s ease',
        textDecoration: 'none',
        '&:hover': {
            color: theme.palette.primary[900],
            textDecoration: 'underline',
        },
    },
    inverse: {
        color: 'rgba(255, 255, 255, 0.9)',
        '&:hover': {
            color: theme.palette.common.white,
        },
    },
    negative: {
        color: theme.palette.negative.main,
        '&:hover': {
            color: theme.palette.negative.dark,
        },
    },
    positive: {
        color: theme.palette.positive.main,
        '&:hover': {
            color: theme.palette.positive.dark,
        },
    },
}); }, { name: LinkStylesKey });
export var Link = function (_a) {
    var _b, _c;
    var to = _a.to, children = _a.children, className = _a.className, _d = _a.color, color = _d === void 0 ? 'default' : _d, newTab = _a.newTab, dataTestId = _a["data-testid"], rootProps = __rest(_a, ["to", "children", "className", "color", "newTab", 'data-testid']);
    var classes = useStyles({});
    if (newTab) {
        return (React.createElement("a", { target: "_blank", rel: "noopener noreferrer", href: to, className: clsx(classes.root, (_b = {},
                _b[classes.inverse] = color === 'inverse',
                _b[classes.negative] = color === 'negative',
                _b[classes.positive] = color === 'positive',
                _b), className), "data-testid": dataTestId, onClick: rootProps.onClick
                ? rootProps.onClick
                : undefined }, children));
    }
    return (React.createElement(NavLink, __assign({ to: to, className: clsx(classes.root, (_c = {},
            _c[classes.inverse] = color === 'inverse',
            _c[classes.negative] = color === 'negative',
            _c[classes.positive] = color === 'positive',
            _c), className), "data-testid": dataTestId }, rootProps), children));
};
