// @itwin/itwinui-react/esm/core/Table/filters/DateRangeFilter/DateRangeFilter.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
/*---------------------------------------------------------------------------------------------
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
import React from 'react';
import { useTheme } from '../../../utils/hooks/useTheme';
import { FilterButtonBar, } from '../FilterButtonBar';
import { BaseFilter } from '../BaseFilter';
import '@itwin/itwinui-css/css/table.css';
import DatePickerInput from './DatePickerInput';
var defaultStrings = {
    from: 'From',
    to: 'To',
};
var defaultFormatDate = function (date) {
    var formatter = new Intl.DateTimeFormat('en-us', {
        month: 'short',
        day: 'numeric',
        year: 'numeric',
    });
    return formatter.format(date);
};
var defaultParseInput = function (text) {
    // Checks whether date is in correct format e.g. Feb 11, 2021
    if (/^[a-z]{3}\s\d{1,2},\s\d{4}$/gi.test(text)) {
        return new Date(text);
    }
    return new Date('');
};
export var DateRangeFilter = function (props) {
    var _a, _b, _c, _d;
    var column = props.column, translatedLabels = props.translatedLabels, setFilter = props.setFilter, clearFilter = props.clearFilter, _e = props.formatDate, formatDate = _e === void 0 ? defaultFormatDate : _e, _f = props.parseInput, parseInput = _f === void 0 ? defaultParseInput : _f, _g = props.placeholder, placeholder = _g === void 0 ? 'MMM dd, yyyy' : _g;
    useTheme();
    var translatedStrings = __assign(__assign({}, defaultStrings), translatedLabels);
    var _h = React.useState(((_a = column.filterValue) === null || _a === void 0 ? void 0 : _a[0]) ? new Date((_b = column.filterValue) === null || _b === void 0 ? void 0 : _b[0]) : undefined), from = _h[0], setFrom = _h[1];
    var _j = React.useState(((_c = column.filterValue) === null || _c === void 0 ? void 0 : _c[1]) ? new Date((_d = column.filterValue) === null || _d === void 0 ? void 0 : _d[1]) : undefined), to = _j[0], setTo = _j[1];
    var onKeyDown = function (event) {
        if (event.key === 'Enter') {
            setFilter([from, to]);
        }
    };
    return (React.createElement(BaseFilter, null,
        React.createElement(DatePickerInput, { label: translatedStrings.from, date: from, onChange: setFrom, formatDate: formatDate, parseInput: parseInput, onKeyDown: onKeyDown, placeholder: placeholder, setFocus: true }),
        React.createElement(DatePickerInput, { label: translatedStrings.to, date: to, onChange: setTo, formatDate: formatDate, parseInput: parseInput, onKeyDown: onKeyDown, placeholder: placeholder }),
        React.createElement(FilterButtonBar, { setFilter: function () { return setFilter([from, to]); }, clearFilter: clearFilter, translatedLabels: translatedLabels })));
};
