// @lifeomic/chroma-react/esm/components/Tabs/Tab.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { Tab as BaseTab } from 'reakit/Tab';
import { makeStyles } from '../../styles';
import { TabsContext } from './TabsContext';
export var TabStylesKey = 'ChromaTab';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        alignItems: 'center',
        backgroundColor: 'transparent',
        border: 'none',
        cursor: 'pointer',
        display: 'inline-flex',
        fontFamily: theme.typography.fontFamily,
        fontSize: theme.typography.subtitle1.fontSize,
        fontWeight: theme.typography.fontWeightRegular,
        justifyContent: 'center',
        letterSpacing: '0.03125em',
        margin: 0,
        minHeight: theme.pxToRem(48),
        outline: 'none',
        overflow: 'hidden',
        paddingBottom: theme.spacing(0.75),
        paddingLeft: theme.spacing(3),
        paddingRight: theme.spacing(3),
        paddingTop: theme.spacing(0.75),
        transition: 'border-bottom 0.25s ease, color 0.5s ease',
        userSelect: 'none',
        verticalAlign: 'middle',
        '&:hover': {
            color: theme.palette.black[500],
        },
        '&:focus': {
            outline: 'none',
        },
        '&[aria-selected="true"]': {
            color: theme.palette.black[900],
            borderBottom: "4px solid " + theme.palette.primary.main,
        },
        '&[aria-disabled="true"]': {
            color: theme.palette.black[400],
            cursor: 'initial',
            opacity: 0.4,
        },
    },
}); }, { name: TabStylesKey });
export var Tab = function (_a) {
    var className = _a.className, disabled = _a.disabled, onClick = _a.onClick, stopId = _a.stopId, rootProps = __rest(_a, ["className", "disabled", "onClick", "stopId"]);
    var classes = useStyles({});
    var tab = React.useContext(TabsContext);
    return (React.createElement(BaseTab, __assign({}, tab, { className: clsx(classes.root, className), disabled: disabled, onClick: onClick, id: stopId }, rootProps)));
};
