// @lifeomic/chroma-react/esm/components/Tabs/TabList.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { TabList as BaseTabList } from 'reakit/Tab';
import { makeStyles } from '../../styles';
import { TabsContext } from './TabsContext';
export var TabListStylesKey = 'ChromaTabList';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        display: 'flex',
        flexDirection: 'row',
        borderBottom: "1px solid " + theme.palette.divider,
        width: '100%',
        overflow: 'hidden',
    },
}); }, { name: TabListStylesKey });
export var TabList = function (_a) {
    var ariaLabel = _a["aria-label"], className = _a.className, rootProps = __rest(_a, ['aria-label', "className"]);
    var classes = useStyles({});
    var tab = React.useContext(TabsContext);
    return (React.createElement(BaseTabList, __assign({}, tab, { "aria-label": ariaLabel, className: clsx(classes.root, className) }, rootProps)));
};
