// @lifeomic/chroma-react/components/Link/Link.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
var react_router_dom_1 = require("react-router-dom");
var styles_1 = require("../../styles");
exports.LinkStylesKey = 'ChromaLink';
exports.useStyles = styles_1.makeStyles(function (theme) { return ({
    root: {
        color: theme.palette.primary.main,
        transition: 'color 0.25s ease',
        textDecoration: 'none',
        '&:hover': {
            color: theme.palette.primary[900],
            textDecoration: 'underline',
        },
    },
    inverse: {
        color: 'rgba(255, 255, 255, 0.9)',
        '&:hover': {
            color: theme.palette.common.white,
        },
    },
}); }, { name: exports.LinkStylesKey });
exports.Link = function (_a) {
    var _b, _c;
    var to = _a.to, children = _a.children, className = _a.className, _d = _a.color, color = _d === void 0 ? 'default' : _d, newTab = _a.newTab, dataTestId = _a["data-testid"], rootProps = __rest(_a, ["to", "children", "className", "color", "newTab", 'data-testid']);
    var classes = exports.useStyles({});
    if (newTab) {
        return (React.createElement("a", { target: "_blank", rel: "noopener noreferrer", href: to, className: clsx_1.default(classes.root, (_b = {}, _b[classes.inverse] = color === 'inverse', _b), className), "data-testid": dataTestId, onClick: rootProps.onClick
                ? rootProps.onClick
                : undefined }, children));
    }
    return (React.createElement(react_router_dom_1.NavLink, __assign({ to: to, className: clsx_1.default(classes.root, (_c = {}, _c[classes.inverse] = color === 'inverse', _c), className), "data-testid": dataTestId }, rootProps), children));
};
