// @lifeomic/chroma-react/components/TableModule/TableActionDivider.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var Divider_1 = require("../Divider");
var styles_1 = require("../../styles");
var React = __importStar(require("react"));
var clsx_1 = __importDefault(require("clsx"));
exports.TableActionDividerStylesKey = 'ChromaTableActionDivider';
exports.useStyles = styles_1.makeStyles(function (theme) { return ({
    root: {
        height: theme.pxToRem(19),
        margin: "0 " + theme.pxToRem(16) + " 0 0",
        top: theme.pxToRem(4),
        position: 'relative',
    },
}); }, { name: exports.TableActionDividerStylesKey });
exports.TableActionDivider = function (_a) {
    var className = _a.className, rootProps = __rest(_a, ["className"]);
    var classes = exports.useStyles({});
    return (React.createElement(Divider_1.Divider, __assign({ className: clsx_1.default(classes.root, className), color: "inverse", direction: "row" }, rootProps)));
};
