// @lifeomic/chroma-react/components/TableModule/TableModuleActions.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var index_1 = require("../../styles/index");
var clsx_1 = __importDefault(require("clsx"));
exports.TableModuleActionsStylesKey = 'ChromaTableModuleActions';
exports.useStyles = index_1.makeStyles(function (theme) { return ({
    root: {
        display: 'flex',
        justifyContent: 'flex-end',
        // Handle the case where an IconButton or IconbuttonLink is in a cell
        // We don't want it to be too large to push out the height of the row!
        '& > button, & > a': {
            display: 'flex',
            height: theme.pxToRem(26),
            marginRight: theme.spacing(2.5),
            padding: 0,
            '& > svg': {
                height: theme.pxToRem(15),
                width: theme.pxToRem(15),
            },
            '&:last-child': {
                marginRight: 0,
            },
        },
    },
}); }, { name: exports.TableModuleActionsStylesKey });
exports.TableModuleActions = function (_a) {
    var children = _a.children, className = _a.className, rootProps = __rest(_a, ["children", "className"]);
    var classes = exports.useStyles({});
    return (React.createElement("div", __assign({ className: clsx_1.default(classes.root, className), "aria-label": "Table Actions" }, rootProps), children));
};
