// @lifeomic/chroma-react/components/Breadcrumbs/BreadcrumbNav.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
var index_1 = require("../../styles/index");
exports.BreadcrumbNavStylesKey = 'ChromaBreadcrumbNav';
exports.useStyles = index_1.makeStyles(function (theme) { return ({
    root: {},
    ol: {
        margin: theme.spacing(0),
        paddingLeft: theme.spacing(0),
        listStyle: 'none',
    },
}); }, { name: exports.BreadcrumbNavStylesKey });
exports.BreadcrumbNav = function (_a) {
    var className = _a.className, children = _a.children, rootProps = __rest(_a, ["className", "children"]);
    var classes = exports.useStyles({});
    return (React.createElement("nav", __assign({ "aria-label": "Breadcrumb", className: clsx_1.default(classes.root, className) }, rootProps),
        React.createElement("ol", { className: classes.ol }, children)));
};
