// @itwin/itwinui-react/cjs/core/Footer/Footer.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __spreadArray = (this && this.__spreadArray) || function (to, from, pack) {
    if (pack || arguments.length === 2) for (var i = 0, l = from.length, ar; i < l; i++) {
        if (ar || !(i in from)) {
            if (!ar) ar = Array.prototype.slice.call(from, 0, i);
            ar[i] = from[i];
        }
    }
    return to.concat(ar || Array.prototype.slice.call(from));
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Footer = void 0;
/*---------------------------------------------------------------------------------------------
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
var react_1 = __importDefault(require("react"));
var classnames_1 = __importDefault(require("classnames"));
var utils_1 = require("../utils");
require("@itwin/itwinui-css/css/footer.css");
var footerTranslations = {
    cookies: 'Cookies',
    legalNotices: 'Legal notices',
    privacy: 'Privacy',
    termsOfService: 'Terms of service',
    termsOfUse: 'Terms of use',
};
/**
 * Footer element with all needed legal and info links.
 * Be sure to place it manually at the bottom of your page.
 * You can use position 'absolute' with relative body or set the height of the content and place footer at the end.
 * @example <caption>Appending custom element after default elements</caption>
 * <Footer customElements={[{title: 'Bentley', url: 'https://www.bentley.com/'}]} />
 * @example <caption>Returning only custom elements</caption>
 * <Footer customElements={() => newFooterElements)} />
 * @example <caption>Filtering out a specific element</caption>
 * <Footer customElements={(defaultElements) => defaultElements.filter(({ key }) => key !== 'privacy' )} />
 * @example <caption>Changing a url</caption>
 * <Footer customElements={(defaultElements) => defaultElements.map(element => ({ ...element, url: element.key === 'privacy' ? customPrivacyUrl : element.url }))} />
 */
var Footer = function (props) {
    var customElements = props.customElements, translatedTitles = props.translatedTitles, className = props.className, rest = __rest(props, ["customElements", "translatedTitles", "className"]);
    (0, utils_1.useTheme)();
    var titles = __assign(__assign({}, footerTranslations), translatedTitles);
    var defaultElements = [
        {
            key: 'copyright',
            title: "\u00A9 ".concat(new Date().getFullYear(), " Bentley Systems, Incorporated"),
        },
        {
            key: 'termsOfService',
            title: titles.termsOfService,
            url: 'https://connect-agreementportal.bentley.com/AgreementApp/Home/Eula/view/readonly/BentleyConnect',
        },
        {
            key: 'privacy',
            title: titles.privacy,
            url: 'https://www.bentley.com/en/privacy-policy',
        },
        {
            key: 'termsOfUse',
            title: titles.termsOfUse,
            url: 'https://www.bentley.com/en/terms-of-use-and-select-online-agreement',
        },
        {
            key: 'cookies',
            title: titles.cookies,
            url: 'https://www.bentley.com/en/cookie-policy',
        },
        {
            key: 'legalNotices',
            title: titles.legalNotices,
            url: 'https://connect.bentley.com/Legal',
        },
    ];
    var elements = defaultElements;
    if (customElements) {
        elements =
            typeof customElements === 'function'
                ? customElements(defaultElements)
                : __spreadArray(__spreadArray([], defaultElements, true), customElements, true);
    }
    return (react_1.default.createElement("footer", __assign({ className: (0, classnames_1.default)('iui-legal-footer', className) }, rest),
        react_1.default.createElement("ul", null, elements.map(function (element, index) {
            return (react_1.default.createElement("li", { key: element.key || "".concat(element.title, "-").concat(index) },
                index > 0 && react_1.default.createElement("span", { className: 'iui-separator' }),
                element.url ? (react_1.default.createElement("a", { href: element.url, target: '_blank', rel: 'noreferrer' }, element.title)) : (element.title)));
        }))));
};
exports.Footer = Footer;
exports.default = exports.Footer;
