// @itwin/itwinui-react/cjs/core/Footer/Footer.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __spreadArray = (this && this.__spreadArray) || function (to, from) {
    for (var i = 0, il = from.length, j = to.length; i < il; i++, j++)
        to[j] = from[i];
    return to;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Footer = void 0;
/*---------------------------------------------------------------------------------------------
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
var react_1 = __importDefault(require("react"));
var classnames_1 = __importDefault(require("classnames"));
var useTheme_1 = require("../utils/hooks/useTheme");
require("@itwin/itwinui-css/css/footer.css");
var footerTranslations = {
    cookies: 'Cookies',
    legalNotices: 'Legal notices',
    privacy: 'Privacy',
    termsOfService: 'Terms of service',
    termsOfUse: 'Terms of use',
};
/**
 * Footer element with all needed legal and info links.
 * Be sure to place it manually at the bottom of your page.
 * You can use position 'absolute' with relative body or set the height of the content and place footer at the end.
 * @example
 * <Footer customElements={[{title: 'Bentley', url: 'https://www.bentley.com/'}]} />
 */
var Footer = function (props) {
    var customElements = props.customElements, translatedTitles = props.translatedTitles, className = props.className, rest = __rest(props, ["customElements", "translatedTitles", "className"]);
    useTheme_1.useTheme();
    var today = new Date();
    var titles = __assign(__assign({}, footerTranslations), translatedTitles);
    var defaultElements = [
        {
            title: titles.termsOfService,
            url: 'https://connect-agreementportal.bentley.com/AgreementApp/Home/Eula/view/readonly/BentleyConnect',
        },
        { title: titles.privacy, url: 'https://www.bentley.com/en/privacy-policy' },
        {
            title: titles.termsOfUse,
            url: 'https://www.bentley.com/en/terms-of-use-and-select-online-agreement',
        },
        { title: titles.cookies, url: 'https://www.bentley.com/en/cookie-policy' },
        { title: titles.legalNotices, url: 'https://connect.bentley.com/Legal' },
    ];
    var elements = customElements
        ? __spreadArray(__spreadArray([], defaultElements), customElements) : defaultElements;
    return (react_1.default.createElement("footer", __assign({ className: classnames_1.default('iui-legal-footer', className) }, rest),
        react_1.default.createElement("ul", null,
            react_1.default.createElement("li", null,
                "\u00A9 ",
                today.getFullYear(),
                " Bentley Systems, Incorporated"),
            elements.map(function (element, index) {
                return (react_1.default.createElement("li", { key: element.title + "-" + index },
                    react_1.default.createElement("span", { className: 'iui-separator' }),
                    element.url ? (react_1.default.createElement("a", { href: element.url, target: '_blank', rel: 'noreferrer' }, element.title)) : (element.title)));
            }))));
};
exports.Footer = Footer;
exports.default = exports.Footer;
