// @lifeomic/chroma-react/esm/components/LinearProgress/LinearProgress.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { makeStyles } from '../../styles';
import { useReducedMotion } from 'framer-motion';
import * as React from 'react';
import clsx from 'clsx';
export var LinearProgressStylesKey = 'ChromaLinearProgress';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        '--linear-progress-height': '0.25rem',
        '--linear-progress-background-color': theme.palette.black[100],
        '--linear-progress-bar-color': "linear-gradient(135deg,\n        ".concat(theme.palette.primary.light, " 0%,\n        ").concat(theme.palette.primary.main, " 22%,\n        ").concat(theme.palette.purple[700], " 92%)"),
        backgroundColor: 'var(--linear-progress-background-color)',
        height: 'var(--linear-progress-height)',
        overflow: 'hidden',
        position: 'relative',
    },
    bar: {
        background: 'var(--linear-progress-bar-color)',
        bottom: 0,
        height: 'var(--linear-progress-height)',
        left: 0,
        position: 'absolute',
        top: 0,
        transition: 'transform 0.4s linear',
        width: '100%',
    },
    indeterminate: {
        animation: '$chroma-progress-bar-indeterminate 2s cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite',
    },
    indeterminateReduced: {
        animation: '$chroma-progress-bar-indeterminate 10s cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite',
    },
    '@keyframes chroma-progress-bar-indeterminate': {
        '0%': {
            left: '-100%',
            right: '100%',
        },
        '60%': {
            left: '50%',
            right: '-50%',
        },
        '100%': {
            left: '100%',
            right: '-100%',
        },
    },
}); }, { name: LinearProgressStylesKey });
export var LinearProgress = function (_a) {
    var className = _a.className, _b = _a.variant, variant = _b === void 0 ? 'determinate' : _b, _c = _a.value, value = _c === void 0 ? 0 : _c, rootProps = __rest(_a, ["className", "variant", "value"]);
    var classes = useStyles({});
    var shouldReduceMotion = useReducedMotion();
    return (React.createElement("div", __assign({ className: clsx(classes.root, className), role: "progressbar", "aria-valuenow": variant === 'determinate' ? value : undefined, "aria-valuemin": variant === 'determinate' ? 0 : undefined, "aria-valuemax": variant === 'determinate' ? 100 : undefined }, rootProps),
        React.createElement("div", { className: clsx(classes.bar, variant === 'indeterminate' && classes.indeterminate, shouldReduceMotion && classes.indeterminateReduced), style: variant === 'determinate'
                ? { transform: "translateX(-".concat(100 - value, "%)") }
                : undefined })));
};
