// @lifeomic/chroma-react/components/FormBox/FormBox.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormBox = exports.useStyles = exports.FormBoxStylesKey = void 0;
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
var styles_1 = require("../../styles");
var Box_1 = require("../Box");
exports.FormBoxStylesKey = 'ChromaFormBox';
exports.useStyles = (0, styles_1.makeStyles)(function (theme) { return ({
    root: {},
    vSpacing0: {
        '& > *:not(:last-child)': {
            marginBottom: 0,
        },
    },
    vSpacing05: {
        '& > *:not(:last-child)': {
            marginBottom: theme.spacing(0.5),
        },
    },
    vSpacing10: {
        '& > *:not(:last-child)': {
            marginBottom: theme.spacing(1),
        },
    },
    vSpacing15: {
        '& > *:not(:last-child)': {
            marginBottom: theme.spacing(1.5),
        },
    },
    vSpacing20: {
        '& > *:not(:last-child)': {
            marginBottom: theme.spacing(2),
        },
    },
    vSpacing25: {
        '& > *:not(:last-child)': {
            marginBottom: theme.spacing(2.5),
        },
    },
    vSpacing30: {
        '& > *:not(:last-child)': {
            marginBottom: theme.spacing(3),
        },
    },
    hSpacing0: {
        '& > *:not(:last-child)': {
            marginRight: 0,
        },
    },
    hSpacing05: {
        '& > *:not(:last-child)': {
            marginRight: theme.spacing(0.5),
        },
    },
    hSpacing10: {
        '& > *:not(:last-child)': {
            marginRight: theme.spacing(1),
        },
    },
    hSpacing15: {
        '& > *:not(:last-child)': {
            marginRight: theme.spacing(1.5),
        },
    },
    hSpacing20: {
        '& > *:not(:last-child)': {
            marginRight: theme.spacing(2),
        },
    },
    hSpacing25: {
        '& > *:not(:last-child)': {
            marginRight: theme.spacing(2.5),
        },
    },
    hSpacing30: {
        '& > *:not(:last-child)': {
            marginRight: theme.spacing(3),
        },
    },
    padding0: {
        padding: 0,
    },
    padding05: {
        padding: theme.spacing(0.5),
    },
    padding10: {
        padding: theme.spacing(1),
    },
    padding15: {
        padding: theme.spacing(1.5),
    },
    padding20: {
        padding: theme.spacing(2),
    },
    xPadding0: {
        paddingLeft: 0,
        paddingRight: 0,
    },
    xPadding05: {
        paddingLeft: theme.spacing(0.5),
        paddingRight: theme.spacing(0.5),
    },
    xPadding10: {
        paddingLeft: theme.spacing(1),
        paddingRight: theme.spacing(0.5),
    },
    xPadding15: {
        paddingLeft: theme.spacing(1.5),
        paddingRight: theme.spacing(0.5),
    },
    xPadding20: {
        paddingLeft: theme.spacing(2),
        paddingRight: theme.spacing(0.5),
    },
    yPadding0: {
        paddingTop: 0,
        paddingBottom: 0,
    },
    yPadding05: {
        paddingTop: theme.spacing(0.5),
        paddingBottom: theme.spacing(0.5),
    },
    yPadding10: {
        paddingTop: theme.spacing(1),
        paddingBottom: theme.spacing(0.5),
    },
    yPadding15: {
        paddingTop: theme.spacing(1.5),
        paddingBottom: theme.spacing(0.5),
    },
    yPadding20: {
        paddingTop: theme.spacing(2),
        paddingBottom: theme.spacing(0.5),
    },
}); }, { name: exports.FormBoxStylesKey });
var FormBox = function (_a) {
    var _b, _c, _d, _e, _f;
    var className = _a.className, children = _a.children, _g = _a.direction, direction = _g === void 0 ? 'column' : _g, _h = _a.spacing, spacing = _h === void 0 ? 2 : _h, padding = _a.padding, paddingY = _a.paddingY, paddingX = _a.paddingX, rootProps = __rest(_a, ["className", "children", "direction", "spacing", "padding", "paddingY", "paddingX"]);
    var classes = (0, exports.useStyles)({});
    return (React.createElement(Box_1.Box, __assign({ direction: direction, className: (0, clsx_1.default)(classes.root, (_b = {},
            _b[classes.vSpacing0] = spacing === 0 && direction === 'column',
            _b[classes.vSpacing05] = spacing === 0.5 && direction === 'column',
            _b[classes.vSpacing10] = spacing === 1 && direction === 'column',
            _b[classes.vSpacing15] = spacing === 1.5 && direction === 'column',
            _b[classes.vSpacing20] = spacing === 2 && direction === 'column',
            _b[classes.vSpacing25] = spacing === 2.5 && direction === 'column',
            _b[classes.vSpacing30] = spacing === 3 && direction === 'column',
            _b), (_c = {},
            _c[classes.hSpacing0] = spacing === 0 && direction === 'row',
            _c[classes.hSpacing05] = spacing === 0.5 && direction === 'row',
            _c[classes.hSpacing10] = spacing === 1 && direction === 'row',
            _c[classes.hSpacing15] = spacing === 1.5 && direction === 'row',
            _c[classes.hSpacing20] = spacing === 2 && direction === 'row',
            _c[classes.hSpacing25] = spacing === 2.5 && direction === 'row',
            _c[classes.hSpacing30] = spacing === 3 && direction === 'row',
            _c), (_d = {},
            _d[classes.padding0] = padding === 0,
            _d[classes.padding05] = padding === 0.5,
            _d[classes.padding10] = padding === 1,
            _d[classes.padding15] = padding === 1.5,
            _d[classes.padding20] = padding === 2,
            _d), (_e = {},
            _e[classes.yPadding0] = paddingY === 0,
            _e[classes.yPadding05] = paddingY === 0.5,
            _e[classes.yPadding10] = paddingY === 1,
            _e[classes.yPadding15] = paddingY === 1.5,
            _e[classes.yPadding20] = paddingY === 2,
            _e), (_f = {},
            _f[classes.xPadding0] = paddingX === 0,
            _f[classes.xPadding05] = paddingX === 0.5,
            _f[classes.xPadding10] = paddingX === 1,
            _f[classes.xPadding15] = paddingX === 1.5,
            _f[classes.xPadding20] = paddingX === 2,
            _f), className) }, rootProps), children));
};
exports.FormBox = FormBox;
