// @lifeomic/chroma-react/esm/components/Menu/MenuButton.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import * as React from 'react';
import { makeStyles } from '../../styles';
import { Button } from '../Button';
export var MenuButtonStylesKey = 'ChromaMenuButton';
export var useStyles = makeStyles(function (theme) { return ({
    trailingIcon: {
        width: theme.spacing(2),
        height: theme.spacing(2),
        marginLeft: theme.spacing(1),
        verticalAlign: 'middle',
        position: 'relative',
        top: theme.pxToRem(-1),
    },
}); }, { name: MenuButtonStylesKey });
/** @deprecated Please use `trailingIcon` from Button instead. This will be removed in a future release */
export var MenuButton = React.forwardRef(function (_a, ref) {
    var children = _a.children, TrailingIcon = _a.trailingIcon, rootProps = __rest(_a, ["children", "trailingIcon"]);
    var classes = useStyles({});
    return (React.createElement(Button, __assign({ ref: ref }, rootProps),
        children,
        !!TrailingIcon && (React.createElement(TrailingIcon, { role: "img", "aria-hidden": true, className: classes.trailingIcon }))));
});
