// @lifeomic/chroma-react/esm/components/Menu/MenuItem.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import clsx from 'clsx';
import * as React from 'react';
import { makeStyles } from '../../styles';
export var MenuItemStylesKey = 'ChromaMenuItem';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        alignItems: 'center',
        backgroundColor: theme.palette.common.white,
        border: 'none',
        cursor: 'pointer',
        display: 'flex',
        fontFamily: theme.typography.fontFamily,
        fontSize: theme.typography.body1.fontSize,
        letterSpacing: '0.015625em',
        margin: 0,
        outline: 'none',
        overflow: 'hidden',
        padding: theme.spacing(1.5, 2.5),
        userSelect: 'none',
        width: '100%',
        '&:hover,&:focus': {
            backgroundColor: 'rgba(222,244,252, 0.6)',
        },
        '&:focus': {
            outline: 'none',
        },
    },
    icon: {
        marginRight: theme.spacing(2),
    },
}); }, { name: MenuItemStylesKey });
export var MenuItem = React.forwardRef(function (_a, ref) {
    var className = _a.className, children = _a.children, Icon = _a.icon, onClick = _a.onClick, text = _a.text, rootProps = __rest(_a, ["className", "children", "icon", "onClick", "text"]);
    var classes = useStyles({});
    var handleStopPropagation = function (e) {
        e.stopPropagation();
        onClick && onClick(e);
    };
    return (React.createElement("button", __assign({ ref: ref, className: clsx(classes.root, className), onClick: handleStopPropagation }, rootProps),
        !!Icon && (React.createElement(Icon, { role: "img", "aria-hidden": true, className: classes.icon, width: 18, height: 18 })),
        !!text && text,
        children));
});
