// @lifeomic/chroma-react/esm/components/NumberFormat/PhoneNumberFormatField.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import 'react-phone-number-input/style.css';
import * as React from 'react';
import { TextField } from '../TextField';
// @ts-ignore The flags module is not typed
import flags from 'react-phone-number-input/flags';
import PhoneInput from 'react-phone-number-input';
import { makeStyles } from '../../styles';
import clsx from 'clsx';
export var PhoneNumberFormatFieldStylesKey = 'ChromaPhoneNumberFormatField';
var useStyles = function (_a) {
    var about = _a.about, errorMessage = _a.errorMessage;
    return makeStyles(function (theme) { return ({
        textField: {
            marginTop: theme.spacing(2),
            marginBottom: theme.spacing(1),
        },
        phoneInputRoot: {
            display: 'flex',
            width: '100%',
            position: 'relative',
            '& > .PhoneInputCountry': {
                position: 'absolute',
                bottom: theme.spacing(about || errorMessage ? 5.25 : 2.5),
                left: theme.spacing(1),
                '& .PhoneInputCountryIcon--border': {
                    boxShadow: 'unset',
                    backgroundColor: 'unset',
                },
            },
        },
        phoneInput: {
            '& > input': {
                paddingLeft: theme.spacing(6),
            },
        },
    }); }, { name: PhoneNumberFormatFieldStylesKey })({});
};
var PhoneInputCompatibleChromaInput = React.forwardRef(function (props, ref) {
    var classes = useStyles({});
    var className = props.className, textFieldProps = __rest(props, ["className"]);
    return (React.createElement(TextField, __assign({ fullWidth: true, label: "Phone Number", helpMessage: props.about, ref: ref }, textFieldProps, { className: clsx(classes.phoneInput, classes.textField, className) })));
});
export var PhoneNumberFormatField = function (_a) {
    var className = _a.className, props = __rest(_a, ["className"]);
    var value = props.value, onChange = props.onChange, errorMessage = props.errorMessage;
    var classes = useStyles(props);
    return (React.createElement(PhoneInput, __assign({ className: clsx(classes.phoneInputRoot, className), defaultCountry: "US", flags: flags }, props, { 
        // @ts-ignore defaultValue has type mismatch
        inputComponent: PhoneInputCompatibleChromaInput, value: value, onChange: onChange, error: errorMessage })));
};
