// @lifeomic/chroma-react/components/Pill/Pill.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
var styles_1 = require("../../styles");
var framer_motion_1 = require("framer-motion");
exports.PillStylesKey = 'ChromaPill';
exports.useStyles = styles_1.makeStyles(function (theme) { return ({
    root: {
        display: 'inline-flex',
        alignItems: 'center',
        background: theme.palette.black[500],
        color: theme.palette.common.white,
        borderRadius: theme.pxToRem(20),
        fontSize: theme.typography.caption.fontSize,
        fontWeight: theme.typography.fontWeightBold,
        padding: theme.spacing(0.5, 1.5),
        width: 'fit-content',
        maxHeight: theme.pxToRem(25),
    },
    default: {},
    primaryColor: {
        background: theme.palette.primary.main,
    },
    highlight: {
        background: "linear-gradient(135deg, " + theme.palette.primary.light + " 0%, " + theme.palette.primary.main + " 40%, rgb(166, 66, 254) 92%)",
        boxShadow: theme.boxShadows.table,
        borderRadius: theme.spacing(0.25, 1),
        borderBottom: 'solid 1px rgba(0, 83, 154, 0.75)',
        borderRight: 'solid 1px rgba(0, 83, 154, 0.75)',
        color: theme.palette.common.white,
        fontSize: theme.pxToRem(9),
        fontWeight: theme.typography.fontWeightBolder,
        height: theme.pxToRem(20),
        justifyContent: 'center',
        letterSpacing: theme.pxToRem(1.6),
        lineHeight: 0.9,
        padding: theme.spacing(0.125, 0.75, 0),
        textIndent: theme.pxToRem(3),
        textTransform: 'uppercase',
    },
}); }, { name: exports.PillStylesKey });
exports.Pill = React.forwardRef(function (_a, ref) {
    var _b, _c;
    var children = _a.children, className = _a.className, color = _a.color, label = _a.label, _d = _a.variant, variant = _d === void 0 ? 'default' : _d, rootProps = __rest(_a, ["children", "className", "color", "label", "variant"]);
    var classes = exports.useStyles({});
    return (React.createElement(framer_motion_1.motion.span, __assign({ ref: ref, className: clsx_1.default(classes.root, (_b = {},
            _b[classes.default] = variant === 'default',
            _b[classes.highlight] = variant === 'highlight',
            _b), (_c = {},
            _c[classes.primaryColor] = color === 'primary',
            _c), className) }, rootProps),
        label,
        children));
});
