// @lifeomic/chroma-react/esm/components/Tooltip/Tooltip.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { fontSizes } from '../../styles/createTypography';
import { makeStyles } from '../../styles';
import { motion, useReducedMotion } from 'framer-motion';
import { Portal } from 'reakit/Portal';
import { Tooltip as ReakitTooltip, TooltipReference, useTooltipState, } from 'reakit/Tooltip';
import clsx from 'clsx';
import * as React from 'react';
export var TooltipStylesKey = 'ChromaTooltip';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        backgroundColor: theme.palette.common.black,
        borderRadius: theme.pxToRem(4),
        boxShadow: theme.boxShadows.tooltip,
        color: theme.palette.common.white,
        fontFamily: theme.typography.fontFamily,
        fontSize: fontSizes.tooltip,
        letterSpacing: '0.021875em',
        maxWidth: theme.pxToRem(500),
        paddingLeft: theme.spacing(1),
        paddingRight: theme.spacing(1),
        paddingTop: theme.spacing(0.5),
        paddingBottom: theme.spacing(0.75),
        zIndex: 1300,
    },
}); }, { name: TooltipStylesKey });
export var Tooltip = function (_a) {
    var children = _a.children, className = _a.className, _b = _a.defaultVisible, defaultVisible = _b === void 0 ? false : _b, _c = _a.placement, placement = _c === void 0 ? 'bottom' : _c, _d = _a.title, title = _d === void 0 ? '' : _d, gutter = _a.gutter, rootProps = __rest(_a, ["children", "className", "defaultVisible", "placement", "title", "gutter"]);
    var classes = useStyles({});
    var shouldReduceMotion = useReducedMotion();
    var tooltip = useTooltipState({
        placement: placement,
        visible: defaultVisible,
        gutter: gutter,
    });
    React.useEffect(function () {
        tooltip.place(placement);
    }, [placement, tooltip]);
    return (React.createElement(React.Fragment, null,
        React.createElement(TooltipReference, __assign({}, tooltip), function (referenceProps) {
            return React.cloneElement(React.Children.only(children), referenceProps);
        }),
        React.createElement(Portal, null,
            React.createElement(ReakitTooltip, __assign({}, tooltip, { as: motion.div, className: clsx(classes.root, className), animate: tooltip.visible
                    ? shouldReduceMotion
                        ? { opacity: 1 }
                        : {
                            opacity: 1,
                            transition: { delay: 0.75 },
                        }
                    : {
                        opacity: 0,
                    } }, rootProps), title))));
};
