// @lifeomic/chroma-react/esm/components/SlideOver/Body.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { makeStyles } from '../../styles';
import * as React from 'react';
import clsx from 'clsx';
export var BodyStylesKey = 'ChromaSlideOverBody';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        display: 'flex',
        padding: theme.spacing(2),
        width: '100%',
        height: '100%',
    },
}); }, { name: BodyStylesKey });
export var Body = function (_a) {
    var as = _a.as, className = _a.className, children = _a.children, rootProps = __rest(_a, ["as", "className", "children"]);
    var classes = useStyles({});
    var AsComponent = as || 'div';
    return (React.createElement(AsComponent, __assign({ className: clsx(classes.root, className) }, rootProps), children));
};
