// @lifeomic/chroma-react/esm/components/SlideOver/Header.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { IconButton } from '../IconButton';
import { makeStyles } from '../../styles';
import { Text } from '../Text';
import { X } from '@lifeomic/chromicons';
import * as React from 'react';
import clsx from 'clsx';
export var HeaderStylesKey = 'ChromaSlideOverHeader';
export var useStyles = makeStyles(function (theme) { return ({
    root: {
        alignItems: 'center',
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'space-between',
        backgroundColor: theme.palette.black[50],
        padding: theme.spacing(2),
    },
}); }, { name: HeaderStylesKey });
export var Header = function (_a) {
    var children = _a.children, additionalClasses = _a.classes, className = _a.className, onClose = _a.onClose, title = _a.title, rootProps = __rest(_a, ["children", "classes", "className", "onClose", "title"]);
    var classes = useStyles({});
    return (React.createElement("div", __assign({ className: clsx(classes.root, additionalClasses === null || additionalClasses === void 0 ? void 0 : additionalClasses.root, className) }, rootProps),
        children,
        title && (React.createElement(Text, { className: clsx(additionalClasses === null || additionalClasses === void 0 ? void 0 : additionalClasses.text), size: "body", weight: "bold" }, title)),
        React.createElement(IconButton, { className: additionalClasses === null || additionalClasses === void 0 ? void 0 : additionalClasses.button, "aria-label": "Close panel", icon: X, onClick: onClose, size: 0 })));
};
