// @lifeomic/chroma-react/components/Avatar/AvatarBadge.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AvatarBadge = exports.useStyles = exports.AvatarBadgeStylesKey = void 0;
var clsx_1 = __importDefault(require("clsx"));
var React = __importStar(require("react"));
var styles_1 = require("../../styles");
var AvatarSizeContext_1 = require("./AvatarSizeContext");
exports.AvatarBadgeStylesKey = 'ChromaAvatarBadge';
exports.useStyles = (0, styles_1.makeStyles)(function (theme) { return ({
    root: {
        alignItems: 'center',
        backgroundColor: theme.palette.red.main,
        border: "".concat(theme.pxToRem(1), " solid ").concat(theme.palette.black[50]),
        borderRadius: 9999,
        bottom: 0,
        color: theme.palette.common.white,
        display: 'flex',
        fontWeight: theme.typography.fontWeightBold,
        justifyContent: 'center',
        position: 'absolute',
        right: 0,
        transform: 'translate(25%, 25%)',
        userSelect: 'none',
    },
    size0: {
        height: theme.pxToRem(12),
        width: theme.pxToRem(12),
    },
    size1: {
        fontSize: theme.pxToRem(8),
        height: theme.pxToRem(16),
        width: theme.pxToRem(16),
    },
    size2: {
        fontSize: theme.pxToRem(10),
        height: theme.pxToRem(20),
        width: theme.pxToRem(20),
    },
}); }, { name: exports.AvatarBadgeStylesKey });
var AvatarBadge = function (_a) {
    var _b;
    var className = _a.className, count = _a.count, isVisible = _a.isVisible, rootProps = __rest(_a, ["className", "count", "isVisible"]);
    var classes = (0, exports.useStyles)({});
    var size = (0, AvatarSizeContext_1.useAvatarSize)().size;
    if (!count && !isVisible) {
        return null;
    }
    return (React.createElement("span", __assign({ "aria-label": count ? "".concat(count, " notifications available") : 'notifications available', role: "status", className: (0, clsx_1.default)(classes.root, (_b = {},
            _b[classes.size0] = size === 0,
            _b[classes.size1] = size === 1,
            _b[classes.size2] = size === 2,
            _b), className) }, rootProps), size !== 0 ? count : null));
};
exports.AvatarBadge = AvatarBadge;
