// @lifeomic/chroma-react/styles/createPalette.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createPalette = void 0;
var deepmerge_1 = __importDefault(require("deepmerge"));
var black_1 = __importDefault(require("../colors/black"));
var blue_1 = __importDefault(require("../colors/blue"));
var charcoal_1 = __importDefault(require("../colors/charcoal"));
var darkGraphite_1 = __importDefault(require("../colors/darkGraphite"));
var graphite_1 = __importDefault(require("../colors/graphite"));
var green_1 = __importDefault(require("../colors/green"));
var grey_1 = __importDefault(require("../colors/grey"));
var navy_1 = __importDefault(require("../colors/navy"));
var orange_1 = __importDefault(require("../colors/orange"));
var purple_1 = __importDefault(require("../colors/purple"));
var red_1 = __importDefault(require("../colors/red"));
var text_1 = __importDefault(require("../colors/text"));
var yellow_1 = __importDefault(require("../colors/yellow"));
var baseMuiPalette = {
    primary: blue_1.default,
    secondary: green_1.default,
    error: red_1.default,
    type: 'light',
    common: {
        black: black_1.default.main,
        white: '#ffffff',
    },
    grey: grey_1.default,
    text: text_1.default,
    divider: grey_1.default[400],
    background: {
        paper: grey_1.default[50],
        default: '#f5f8fa',
    },
    negative: {
        main: red_1.default.main,
        dark: red_1.default.dark,
    },
    positive: {
        main: green_1.default.main,
        dark: green_1.default.dark,
    },
};
var basePalette = __assign({ black: black_1.default, blue: blue_1.default, charcoal: charcoal_1.default, darkGraphite: darkGraphite_1.default, graphite: graphite_1.default, green: green_1.default, navy: navy_1.default, orange: orange_1.default, purple: purple_1.default, red: red_1.default, yellow: yellow_1.default }, baseMuiPalette);
var createPalette = function (palette) {
    if (palette === void 0) { palette = {}; }
    return (0, deepmerge_1.default)(basePalette, palette, { clone: false });
};
exports.createPalette = createPalette;
