// @lifeomic/chroma-react/styles/createBoxShadows.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createBoxShadows = exports.baseBoxShadows = void 0;
exports.baseBoxShadows = {
    table: '0px 2px 8px rgba(0, 0, 0, 0.13)',
    header: '0px 0px 24px rgba(0, 0, 0, 0.15)',
    elevatedContent: '0px 10px 64px rgba(0, 0, 0, 0.15)',
    popover: '0px 10px 64px rgba(0, 0, 0, 0.2)',
    tooltip: '0px 0px 24px rgba(0, 0, 0, 0.15)',
    focusVisible: '0 0 0 2px rgba(0, 150, 225, .3)',
    focusVisibleInverse: '0 0 0 2px rgba(255, 255, 255, .3)',
};
var createBoxShadows = function (shadows) {
    if (shadows === void 0) { shadows = {}; }
    return (__assign(__assign({}, exports.baseBoxShadows), shadows));
};
exports.createBoxShadows = createBoxShadows;
