// @lifeomic/chroma-react/styles/createTypography.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createTypography = exports.fontSizes = exports.fontWeights = exports.fontFamilies = void 0;
var deepmerge_1 = __importDefault(require("deepmerge"));
exports.fontFamilies = {
    sansSerif: '"Lato", Helvetica, Roboto, Arial, "Lucida Grande", sans-serif',
    monospace: '"Ubuntu Mono", Consolas, "Lucida Console", Monaco, Courier, monospace',
};
exports.fontWeights = {
    lighter: 100,
    light: 300,
    regular: 400,
    bold: 700,
    bolder: 900,
};
exports.fontSizes = {
    h1: '3rem',
    h2: '2.25rem',
    h3: '1.875rem',
    h4: '1.5rem',
    h5: '1.25rem',
    h6: '1.125rem',
    body: '1rem',
    button: '0.75rem',
    subtitle: '0.875rem',
    caption: '0.75rem',
    tooltip: '0.6875rem',
    code: '1rem',
};
var baseTypography = {
    fontFamily: exports.fontFamilies.sansSerif,
    fontFamilyMonospace: exports.fontFamilies.monospace,
    fontWeightLighter: exports.fontWeights.lighter,
    fontWeightLight: exports.fontWeights.light,
    fontWeightRegular: exports.fontWeights.regular,
    fontWeightMedium: exports.fontWeights.regular,
    fontWeightBold: exports.fontWeights.bold,
    fontWeightBolder: exports.fontWeights.bolder,
    h1: {
        fontSize: exports.fontSizes.h1,
    },
    h2: {
        fontSize: exports.fontSizes.h2,
    },
    h3: {
        fontSize: exports.fontSizes.h3,
    },
    h4: {
        fontSize: exports.fontSizes.h4,
    },
    h5: {
        fontSize: exports.fontSizes.h5,
    },
    h6: {
        fontSize: exports.fontSizes.h6,
    },
    body1: {
        fontSize: exports.fontSizes.body,
    },
    body2: {
        fontSize: exports.fontSizes.subtitle,
    },
    subtitle1: {
        fontSize: exports.fontSizes.subtitle,
    },
    subtitle2: {
        fontSize: exports.fontSizes.subtitle,
    },
    button: {
        textTransform: 'initial',
        fontSize: exports.fontSizes.button,
        fontWeight: exports.fontWeights.bold,
        letterSpacing: 0.5,
        lineHeight: 1.4, // Improve readability for multiline button.
    },
    caption: {
        fontSize: exports.fontSizes.caption,
    },
};
var createTypography = function (typography) {
    if (typography === void 0) { typography = {}; }
    return (0, deepmerge_1.default)(baseTypography, typography, { clone: false });
};
exports.createTypography = createTypography;
