// @lifeomic/chroma-react/styles/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ThemeProvider_1 = require("@material-ui/styles/ThemeProvider");
exports.ThemeProvider = ThemeProvider_1.default;
var createTheme_1 = require("./createTheme");
exports.createTheme = createTheme_1.createTheme;
exports.useTheme = createTheme_1.useTheme;
var createPalette_1 = require("./createPalette");
exports.createPalette = createPalette_1.createPalette;
var createTypography_1 = require("./createTypography");
exports.createTypography = createTypography_1.createTypography;
exports.fontSizes = createTypography_1.fontSizes;
exports.fontWeights = createTypography_1.fontWeights;
var makeStyles_1 = require("./makeStyles");
exports.makeStyles = makeStyles_1.makeStyles;
var createStyles_1 = require("./createStyles");
exports.createStyles = createStyles_1.createStyles;
var withStyles_1 = require("./withStyles");
exports.withStyles = withStyles_1.withStyles;
var withTheme_1 = require("./withTheme");
exports.withTheme = withTheme_1.withTheme;
var useMediaQuery_1 = require("./useMediaQuery");
exports.useMediaQuery = useMediaQuery_1.useMediaQuery;
